// maz-ui/components/MazBadge.mjs
import "./assets/MazBadge.css";
import { defineComponent as l, openBlock as r, createElementBlock as d, normalizeClass as s, normalizeStyle as i, renderSlot as u } from "vue";
const c = /* @__PURE__ */ l({
  __name: "MazBadge",
  props: {
    color: {
      type: String,
      default: "primary",
      validator: (e) => [
        "primary",
        "secondary",
        "warning",
        "danger",
        "info",
        "success",
        "white",
        "gray",
        "black"
      ].includes(e)
    },
    size: { type: String, default: "0.8em" },
    nowrap: { type: Boolean, default: !1 },
    outline: { type: Boolean, default: !1 },
    pastel: { type: Boolean, default: !1 },
    roundedSize: {
      type: String,
      default: "md",
      validator: (e) => ["none", "sm", "md", "lg", "xl", "full"].includes(e)
    }
  },
  setup(e) {
    return (t, a) => (r(), d(
      "span",
      {
        class: s(["m-badge", [
          `--${e.color}`,
          { "--outline": e.outline, "--pastel": e.pastel, "--nowrap": e.nowrap },
          `--rounded-${e.roundedSize}`
        ]]),
        style: i({ fontSize: e.size })
      },
      [
        u(t.$slots, "default", {}, void 0, !0)
      ],
      6
      /* CLASS, STYLE */
    ));
  }
}), f = (e, t) => {
  const a = e.__vccOpts || e;
  for (const [n, o] of t)
    a[n] = o;
  return a;
}, y = /* @__PURE__ */ f(c, [["__scopeId", "data-v-8368c7f0"]]);
export {
  y as default
};
