// maz-ui/components/MazCardSpotlight.mjs
import "./assets/MazCardSpotlight.css";
import { defineComponent as _, useCssVars as b, ref as f, computed as p, onMounted as h, onUnmounted as k, openBlock as C, createElementBlock as E, normalizeClass as g, normalizeStyle as v, createElementVNode as c, renderSlot as y } from "vue";
const z = { class: "inner" }, w = { class: "content" }, B = /* @__PURE__ */ _({
  __name: "MazCardSpotlight",
  props: {
    noElevation: { type: Boolean, default: !1 },
    color: { default: "primary" }
  },
  setup(l) {
    b((e) => ({
      "7cbdd4c5": i.value,
      "4d1dc447": d.value
    }));
    const a = l, t = f(), n = f();
    function r(e) {
      var m, s, u;
      const o = (m = n.value) == null ? void 0 : m.getBoundingClientRect();
      o && ((u = (s = t.value) == null ? void 0 : s.animate) == null || u.call(
        s,
        [
          {
            transform: `translate(${e.clientX - o.left - o.width / 2}px,${e.clientY - o.top - o.height / 2}px)`
          }
        ],
        {
          duration: 300,
          fill: "forwards"
        }
      ));
    }
    const d = p(() => `var(--maz-color-${a.color}-alpha)`), i = p(() => `var(--maz-color-${a.color}-alpha-20)`);
    return h(() => {
      window.addEventListener("mousemove", r);
    }), k(() => {
      window.removeEventListener("mousemove", r);
    }), (e, o) => (C(), E(
      "div",
      {
        class: g(["card", { "maz-elevation": !e.noElevation }]),
        style: v({ backgroundColor: i.value })
      },
      [
        c("div", z, [
          c("div", w, [
            y(e.$slots, "default", {}, void 0, !0)
          ])
        ]),
        c(
          "div",
          {
            ref_key: "blobElement",
            ref: t,
            class: "blob",
            style: v({ backgroundColor: d.value })
          },
          null,
          4
          /* STYLE */
        ),
        c(
          "div",
          {
            ref_key: "fakeblobElement",
            ref: n,
            class: "fakeblob"
          },
          null,
          512
          /* NEED_PATCH */
        )
      ],
      6
      /* CLASS, STYLE */
    ));
  }
}), $ = (l, a) => {
  const t = l.__vccOpts || l;
  for (const [n, r] of a)
    t[n] = r;
  return t;
}, M = /* @__PURE__ */ $(B, [["__scopeId", "data-v-fe909461"]]);
export {
  M as default
};
