// maz-ui/components/MazDropdown.mjs
import "./assets/MazDropdown.css";
import { computed as K, defineComponent as P, defineAsyncComponent as M, getCurrentInstance as G, ref as B, watch as E, openBlock as p, createElementBlock as k, unref as u, normalizeClass as v, normalizeStyle as J, createElementVNode as b, withModifiers as A, createCommentVNode as c, toDisplayString as $, renderSlot as w, createVNode as O, mergeProps as Q, withCtx as h, createBlock as I, Transition as W, withDirectives as X, Fragment as S, renderList as Y, resolveDynamicComponent as Z, createTextVNode as x, vShow as _ } from "vue";
const ee = ({
  componentName: f,
  instance: i,
  providedId: d
}) => K(() => d ?? `${f}-${i == null ? void 0 : i.uid}`);
function oe(f, i) {
  let d;
  return function(...y) {
    clearTimeout(d), d = setTimeout(() => {
      f.apply(this, y);
    }, i);
  };
}
const ne = ["id"], te = ["aria-expanded", "onClick"], le = { class: "maz-sr-only" }, re = { class: "maz-flex maz-items-center maz-gap-2" }, ae = /* @__PURE__ */ P({
  inheritAttrs: !1,
  __name: "MazDropdown",
  props: {
    style: { default: void 0 },
    class: { default: void 0 },
    items: {},
    open: { type: Boolean },
    id: { default: "maz-dropdown" },
    trigger: { default: "both" },
    color: { default: "transparent" },
    position: { default: "bottom left" },
    noCloseOnClick: { type: Boolean },
    disabled: { type: Boolean },
    noChevron: { type: Boolean },
    screenReaderDescription: { default: "Menu Dropdown" }
  },
  emits: [
    /**
     * emitted when a menu item is clicked
     */
    "menuitem-clicked",
    /**
     * Triggers when the number changes
     *
     * @property {boolean} open new value
     */
    "update:open"
  ],
  setup(f, { emit: i }) {
    const d = M(() => import("./chunks/MazBtn-1wfj_m47.mjs").then((e) => e.M)), y = M(() => import("./chunks/chevron-down-bC0s_0yH.mjs")), l = f, C = i, D = ee({
      componentName: "MazCheckbox",
      instance: G(),
      providedId: l.id
    }), r = B(l.open), t = B();
    function L() {
      s(!r.value);
    }
    function N() {
      ["click"].includes(l.trigger) && L();
    }
    function F() {
      ["hover", "both"].includes(l.trigger) && s(!0);
    }
    function T() {
      ["hover", "both"].includes(l.trigger) && (r.value === !1 ? s(!0) : a(!0));
    }
    function U() {
      ["hover", "both"].includes(l.trigger) && a(!1);
    }
    function V() {
      a(!1);
    }
    const s = (e) => {
      l.disabled || (r.value = e, C("update:open", e));
    }, a = oe((e) => {
      s(e);
    }, 200);
    async function j(e, n) {
      var o;
      C("menuitem-clicked", n), await ((o = e.action) == null ? void 0 : o.call(e)), l.noCloseOnClick || g();
    }
    function g() {
      l.noCloseOnClick === !1 && s(!1);
    }
    function z(e) {
      e.key === "Escape" ? (e.preventDefault(), s(!1)) : ["ArrowDown", "ArrowUp"].includes(e.key) ? H(e) : e.key === "Enter" && typeof t.value == "number" && (e.preventDefault(), document.querySelectorAll(`#${D.value} .menuitem`)[t.value].click(), g());
    }
    function q(e) {
      ["ArrowDown", "ArrowUp", "Enter"].includes(e.key) && r.value === !1 && (e.preventDefault(), s(!0));
    }
    function H(e) {
      var m;
      e.preventDefault();
      const n = e.key;
      r.value || s(!0);
      const o = (m = l.items) == null ? void 0 : m.length;
      o && (typeof t.value == "number" ? t.value === o - 1 && n === "ArrowDown" ? t.value = 0 : t.value === 0 && n === "ArrowUp" ? t.value = o - 1 : t.value = n === "ArrowDown" ? t.value + 1 : t.value - 1 : t.value = n === "ArrowDown" ? 0 : o - 1);
    }
    return E(
      () => r.value,
      (e) => {
        e ? document.addEventListener("keydown", z) : document.removeEventListener("keydown", z), t.value = void 0;
      }
    ), E(
      () => l.open,
      (e) => s(e)
    ), (e, n) => (p(), k("div", {
      id: u(D),
      class: v(["m-dropdown", [l.class]]),
      style: J(e.style)
    }, [
      b("div", {
        role: "button",
        tabindex: "0",
        class: "maz-h-full maz-w-full maz-outline-none",
        "aria-expanded": r.value,
        "aria-haspopup": "menu",
        onClick: A(N, ["stop"]),
        onFocus: F,
        onBlur: V,
        onKeydown: q,
        onMouseenter: T,
        onMouseleave: U
      }, [
        c(`
          @slot Custom Element
            @binding {Boolen} is-open close function
            @default \`<MazBtn />\`
        `),
        b(
          "span",
          le,
          $(e.screenReaderDescription),
          1
          /* TEXT */
        ),
        w(e.$slots, "element", { isOpen: r.value }, () => [
          O(u(d), Q({
            color: e.color,
            disabled: e.disabled
          }, e.$attrs, { tabindex: "-1" }), {
            default: h(() => [
              b("span", re, [
                c(" @slot Button text "),
                w(e.$slots, "default"),
                e.noChevron ? c("v-if", !0) : (p(), I(u(y), {
                  key: 0,
                  class: v([{ "maz-rotate-180": r.value }, "maz-text-lg maz-transition-transform maz-duration-200 maz-ease-in-out"])
                }, null, 8, ["class"]))
              ]),
              c(" @slot Menu Label ")
            ]),
            _: 3
            /* FORWARDED */
          }, 16, ["color", "disabled"])
        ])
      ], 40, te),
      O(W, {
        name: "maz-scale-fade",
        persisted: ""
      }, {
        default: h(() => [
          X(b(
            "div",
            {
              role: "menu",
              "aria-label": "Menu",
              class: v(["menu", {
                "--top": e.position.includes("top"),
                "--left": e.position.includes("left"),
                "--right": e.position.includes("right"),
                "--bottom": e.position.includes("bottom")
              }]),
              tabindex: "-1",
              onFocus: n[0] || (n[0] = (o) => u(a)(!0)),
              onBlur: n[1] || (n[1] = (o) => u(a)(!1)),
              onMouseenter: n[2] || (n[2] = (o) => ["hover", "both"].includes(e.trigger) ? u(a)(!0) : void 0),
              onMouseleave: n[3] || (n[3] = (o) => ["hover", "both"].includes(e.trigger) ? u(a)(!1) : void 0)
            },
            [
              (p(!0), k(
                S,
                null,
                Y(e.items, (o, m) => (p(), k(
                  S,
                  null,
                  [
                    c(`
            @slot Custom menu item
              @binding {Object} item - menu item
          `),
                    w(e.$slots, "menuitem", { item: o }, () => [
                      (p(), I(Z(o.to ? "router-link" : o.action ? "button" : "a"), {
                        key: m,
                        href: o.href,
                        tabindex: "-1",
                        target: o.href ? o.target ?? "_self" : void 0,
                        type: o.action ? "button" : void 0,
                        class: v([[
                          {
                            "--is-keyboard-selected": t.value === m
                          },
                          o.class
                        ], "menuitem"]),
                        to: o.to,
                        onClick: A((R) => o.action ? j(o, R) : g(), ["stop"])
                      }, {
                        default: h(() => [
                          c(`
                @slot Custom menu item label
                  @binding {Object} item - menu item
              `),
                          w(e.$slots, "menuitem-label", { item: o }, () => [
                            x(
                              $(o.label),
                              1
                              /* TEXT */
                            )
                          ])
                        ]),
                        _: 2
                        /* DYNAMIC */
                      }, 1032, ["href", "target", "type", "class", "to", "onClick"]))
                    ])
                  ],
                  64
                  /* STABLE_FRAGMENT */
                ))),
                256
                /* UNKEYED_FRAGMENT */
              ))
            ],
            34
            /* CLASS, NEED_HYDRATION */
          ), [
            [_, r.value]
          ])
        ]),
        _: 3
        /* FORWARDED */
      })
    ], 14, ne));
  }
});
export {
  ae as default
};
