// maz-ui/components/MazTransitionExpand.mjs
import "./assets/MazTransitionExpand.css";
import { defineComponent as p, useCssVars as d, openBlock as h, createBlock as u, TransitionGroup as y, withCtx as g, renderSlot as l, nextTick as e } from "vue";
const f = /* @__PURE__ */ p({
  __name: "MazTransitionExpand",
  props: {
    animationDuration: { default: "300ms" }
  },
  setup(n) {
    d((t) => ({
      "3fc44521": t.animationDuration
    }));
    const a = (t) => {
      const o = getComputedStyle(t).width;
      t.style.width = o, t.style.position = "absolute", t.style.visibility = "hidden", t.style.height = "auto", t.style.top = "bottom";
      const i = getComputedStyle(t).height;
      t.style.width = "", t.style.position = "", t.style.visibility = "", t.style.height = "0px", getComputedStyle(t).height, e(() => {
        t.style.height = i;
      });
    }, s = (t) => {
      t.style.height = "auto";
    }, r = (t) => {
      const o = getComputedStyle(t).height;
      t.style.height = o, getComputedStyle(t).height, e(() => {
        t.style.height = "0px";
      });
    };
    return (t, o) => (h(), u(y, {
      class: "m-transition-expand",
      name: "maz-expand",
      tag: "div",
      onEnter: o[0] || (o[0] = (i) => a(i)),
      onAfterEnter: o[1] || (o[1] = (i) => s(i)),
      onLeave: o[2] || (o[2] = (i) => r(i))
    }, {
      default: g(() => [
        l(t.$slots, "default", {}, void 0, !0)
      ]),
      _: 3
      /* FORWARDED */
    }));
  }
}), c = (n, a) => {
  const s = n.__vccOpts || n;
  for (const [r, t] of a)
    s[r] = t;
  return s;
}, C = /* @__PURE__ */ c(f, [["__scopeId", "data-v-77b8b162"]]);
export {
  C as default
};
