// maz-ui/components/MazTabsBar.mjs
import "./assets/MazTabsBar.css";
import { inject as k, defineComponent as $, ref as b, computed as y, onBeforeMount as B, onMounted as M, openBlock as f, createElementBlock as d, normalizeClass as v, createElementVNode as P, normalizeStyle as h, Fragment as C, renderList as q, unref as E, toDisplayString as L } from "vue";
function R(a, t) {
  const o = k(a, t);
  if (!o)
    throw new TypeError(`[maz-ui](injectStrict) Could not resolve ${a}`);
  return o;
}
const j = ["disabled", "onClick"], U = /* @__PURE__ */ $({
  __name: "MazTabsBar",
  props: {
    items: { type: Array, required: !0 },
    persistent: { type: Boolean, default: !1 },
    queryParam: { type: String, default: "tab" },
    color: { type: String, default: "primary" },
    block: { type: Boolean, default: !1 }
  },
  setup(a) {
    const { currentTab: t, updateCurrentTab: o } = R("maz-tabs");
    function c(e) {
      o(e), n.persistent && z(e);
    }
    const n = a, _ = b(), p = b([]);
    function g({
      mazBtn: e,
      index: r
    }) {
      p.value[r] = e && "$el" in e ? e.$el : e;
    }
    const u = y(
      () => n.items.map((e) => ({
        label: typeof e == "string" ? e : e.label,
        disabled: typeof e == "string" ? !1 : e.disabled ?? !1
      }))
    ), T = y(() => {
      if (typeof t.value != "number")
        return {};
      const e = p.value[t.value - 1], r = (e == null ? void 0 : e.offsetWidth) ?? 0, i = (e == null ? void 0 : e.offsetHeight) ?? 0;
      return {
        transform: `translateX(${(e == null ? void 0 : e.offsetLeft) ?? 0}px)`,
        width: `${r}px`,
        height: `${i}px`
      };
    });
    function w(e, r) {
      return r ? {} : t.value === e ? "color: var(--maz-color-text)" : "color: var(--maz-color-muted)";
    }
    B(() => {
      (t.value < 1 || t.value > u.value.length) && console.error(
        `[maz-ui](MazTabsBar) The model-value should be between 1 and ${u.value.length}`
      );
    });
    function S() {
      const e = new URL(window.location.href);
      return Number(e.searchParams.get(n.queryParam));
    }
    function z(e) {
      const r = new URL(window.location.href);
      r.searchParams.set(n.queryParam, String(e)), window.history.replaceState({}, document.title, r.toString());
    }
    return M(async () => {
      n.persistent && o(S() || t.value || 1);
    }), (e, r) => (f(), d(
      "div",
      {
        ref_key: "MazTabsBar",
        ref: _,
        class: v(["m-tabs-bar", {
          "--block": a.block
        }])
      },
      [
        P(
          "div",
          {
            class: "m-tabs-bar__indicator",
            style: h([T.value])
          },
          null,
          4
          /* STYLE */
        ),
        (f(!0), d(
          C,
          null,
          q(u.value, ({ label: i, disabled: l }, s) => (f(), d("button", {
            key: s,
            ref_for: !0,
            ref: (m) => g({ mazBtn: m, index: s }),
            class: v([{ "--active": E(t) === s, "--disabled": l }, "m-tabs-bar__item"]),
            disabled: l,
            style: h(w(s + 1, l)),
            onClick: (m) => l ? void 0 : c(s + 1)
          }, L(i), 15, j))),
          128
          /* KEYED_FRAGMENT */
        ))
      ],
      2
      /* CLASS */
    ));
  }
}), V = (a, t) => {
  const o = a.__vccOpts || a;
  for (const [c, n] of t)
    o[c] = n;
  return o;
}, H = /* @__PURE__ */ V(U, [["__scopeId", "data-v-25e7a5dc"]]);
export {
  H as default
};
