// maz-ui/components/chunks/MazAvatar-0JzIvOc4.mjs
import "../assets/MazAvatar.css";
import { defineComponent as B, defineAsyncComponent as f, computed as m, openBlock as r, createBlock as v, resolveDynamicComponent as w, normalizeStyle as y, normalizeClass as k, withCtx as $, createElementVNode as b, unref as g, createCommentVNode as o, renderSlot as d, toDisplayString as C, createElementBlock as h, createVNode as I } from "vue";
const x = ["tabindex"], L = { class: "m-avatar__initial" }, E = {
  key: 0,
  class: "m-avatar__caption"
}, M = /* @__PURE__ */ B({
  __name: "MazAvatar",
  props: {
    src: {
      type: String,
      default: void 0
    },
    caption: {
      type: String,
      default: void 0
    },
    href: { type: String, default: void 0 },
    to: { type: Object, default: void 0 },
    alt: { type: String, default: "avatar image" },
    target: { type: String, default: "_self" },
    size: { type: String, default: void 0 },
    bordered: { type: Boolean, default: !1 },
    clickable: { type: Boolean, default: !1 },
    square: { type: Boolean, default: !1 },
    noElevation: { type: Boolean, default: !1 },
    showCaption: { type: Boolean, default: !1 },
    imageHeightFull: { type: Boolean, default: !1 },
    noLoader: { type: Boolean, default: !1 },
    buttonColor: {
      type: String,
      default: "info"
    },
    /** Remove the icon on hover when component is clickable */
    noClickableIcon: { type: Boolean, default: !1 },
    letterCount: { type: Number, default: void 0 },
    roundedSize: {
      type: String,
      default: "full",
      validator: (e) => ["none", "sm", "md", "lg", "xl", "full"].includes(e)
    }
  },
  emits: ["click"],
  setup(e) {
    const c = f(() => import("./MazLazyImg-evGMN9Pe.mjs")), n = f(() => import("./pencil-laNW5qNQ.mjs")), t = e, s = m(() => t.to ? "RouterLink" : t.href ? "a" : "div"), u = m(() => !!t.to || !!t.href);
    function z(a, l = t.letterCount) {
      return a.split(" ").map((S) => S[0]).join("").slice(0, l);
    }
    return (a, l) => (r(), v(w(s.value), {
      style: y({ fontSize: e.size }),
      class: k(["m-avatar", [
        {
          "--has-link": u.value
        }
      ]]),
      href: e.href,
      to: e.to,
      target: u.value ? e.target : void 0
    }, {
      default: $(() => [
        b("div", {
          class: k(["m-avatar__wrapper", [
            {
              "--has-shadow": !e.noElevation,
              "--bordered": e.bordered,
              "--clickable": e.clickable,
              "--has-initial": !e.src && e.caption
            },
            `--rounded-${e.square ? "none" : e.roundedSize}`
          ]]),
          tabindex: e.clickable ? 0 : -1
        }, [
          e.src ? (r(), v(g(c), {
            key: 0,
            class: "m-avatar__picture maz-max-w-full",
            image: e.src,
            alt: e.alt,
            "image-height-full": "",
            "no-loader": e.noLoader,
            onClick: l[0] || (l[0] = (i) => e.clickable ? a.$emit("click", i) : null)
          }, null, 8, ["image", "alt", "no-loader"])) : o("v-if", !0),
          e.caption && !e.src ? d(a.$slots, "round-text", { key: 1 }, () => [
            b(
              "span",
              L,
              C(z(e.caption)),
              1
              /* TEXT */
            )
          ], !0) : o("v-if", !0),
          e.clickable ? (r(), h(
            "button",
            {
              key: 2,
              type: "button",
              tabindex: "-1",
              class: "m-avatar__button",
              style: y({
                backgroundColor: e.src ? `var(--maz-color-${e.buttonColor}-alpha)` : `var(--maz-color-${e.buttonColor})`
              }),
              onClick: l[1] || (l[1] = (i) => a.$emit("click", i))
            },
            [
              e.noClickableIcon ? o("v-if", !0) : d(a.$slots, "icon", { key: 0 }, () => [
                I(g(n), { class: "m-avatar__button__icon" })
              ], !0)
            ],
            4
            /* STYLE */
          )) : o("v-if", !0)
        ], 10, x),
        d(a.$slots, "caption", {}, () => [
          e.showCaption && e.caption ? (r(), h(
            "p",
            E,
            C(e.caption),
            1
            /* TEXT */
          )) : o("v-if", !0)
        ], !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["style", "class", "href", "to", "target"]));
  }
}), N = (e, c) => {
  const n = e.__vccOpts || e;
  for (const [t, s] of c)
    n[t] = s;
  return n;
}, q = /* @__PURE__ */ N(M, [["__scopeId", "data-v-1cb092ea"]]);
export {
  q as M,
  N as _
};
