// maz-ui/components/chunks/MazBottomSheet-lN1FCdEJ.mjs
import "../assets/MazBottomSheet.css";
import { defineComponent as y, ref as _, onMounted as w, watch as M, openBlock as c, createBlock as p, Teleport as $, createVNode as B, Transition as L, withCtx as u, createElementBlock as h, normalizeClass as f, createElementVNode as d, withModifiers as z, mergeProps as O, renderSlot as E, createCommentVNode as k, defineAsyncComponent as V, unref as C, pushScopeId as D, popScopeId as P } from "vue";
const g = "--backdrop-present", N = () => {
  document.documentElement.classList.add(g);
}, I = async () => {
  document.querySelector(".m-backdrop.--present") || document.documentElement.classList.remove(g);
}, T = y({
  inheritAttrs: !1,
  props: {
    modelValue: { type: Boolean, default: !1 },
    teleportSelector: { type: String, default: "body" },
    beforeClose: {
      // eslint-disable-next-line @typescript-eslint/ban-types
      type: Function,
      default: void 0
    },
    persistent: { type: Boolean, default: !1 },
    noCloseOnEscKey: { type: Boolean, default: !1 },
    transitionName: { type: String, default: "backdrop-anim" },
    backdropClass: { type: [Array, String, Object], default: void 0 },
    backdropContentClass: {
      type: [Array, String, Object],
      default: void 0
    }
  },
  emits: ["open", "before-close", "close", "update:model-value"],
  setup(e, { emit: t }) {
    const n = _(e.modelValue), l = () => {
      s(!1);
    }, s = async (r) => {
      var b;
      r || (t("before-close"), await ((b = e.beforeClose) == null ? void 0 : b.call(e))), n.value = r;
    }, a = () => {
      t("open");
    }, o = () => {
      t("update:model-value", !1), t("close"), m();
    }, S = () => {
      e.persistent || l();
    }, i = (r) => {
      !e.noCloseOnEscKey && r.key === "Escape" && !e.persistent && l();
    }, v = () => {
      N(), document.addEventListener("keyup", i, !1);
    }, m = () => {
      document.removeEventListener("keyup", i), I();
    };
    return w(() => {
      e.modelValue ? v() : m();
    }), M(
      () => e.modelValue,
      (r) => {
        n.value = r, r ? v() : m();
      }
    ), {
      onBackdropAnimationEnter: a,
      onBackdropAnimationLeave: o,
      onBackdropClicked: S,
      close: l,
      present: n,
      toggleModal: s,
      onKeyPress: i
    };
  }
}), A = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [l, s] of t)
    n[l] = s;
  return n;
};
function j(e, t, n, l, s, a) {
  return c(), p($, { to: e.teleportSelector }, [
    B(L, {
      appear: "",
      name: e.transitionName,
      onAfterEnter: e.onBackdropAnimationEnter,
      onAfterLeave: e.onBackdropAnimationLeave
    }, {
      default: u(() => [
        e.present ? (c(), h(
          "div",
          {
            key: 0,
            class: f(["m-backdrop --present", [e.backdropClass]]),
            tabindex: "-1",
            role: "dialog"
          },
          [
            d(
              "button",
              {
                class: f(["m-backdrop-overlay", { "--disabled": e.persistent }]),
                tabindex: "-1",
                onClick: t[0] || (t[0] = z((...o) => e.onBackdropClicked && e.onBackdropClicked(...o), ["self"]))
              },
              null,
              2
              /* CLASS */
            ),
            d(
              "div",
              O({
                class: ["m-backdrop-content", e.backdropContentClass]
              }, e.$attrs, {
                role: "document",
                tabindex: "0"
              }),
              [
                E(e.$slots, "default", { close: e.close })
              ],
              16
              /* FULL_PROPS */
            )
          ],
          2
          /* CLASS */
        )) : k("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["name", "onAfterEnter", "onAfterLeave"])
  ], 8, ["to"]);
}
const K = /* @__PURE__ */ A(T, [["render", j]]), F = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 24 24"
}, q = /* @__PURE__ */ d(
  "path",
  {
    stroke: "currentColor",
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "1.5",
    d: "M6 18 18 6M6 6l12 12"
  },
  null,
  -1
  /* HOISTED */
), U = [
  q
];
function X(e, t) {
  return c(), h("svg", F, [...U]);
}
const G = { render: X }, H = (e) => (D("data-v-46b34248"), e = e(), P(), e), J = /* @__PURE__ */ H(() => /* @__PURE__ */ d(
  "div",
  { class: "m-bottom-sheet__content-wrapper" },
  [
    /* @__PURE__ */ d("p", null, "Default content")
  ],
  -1
  /* HOISTED */
)), Q = /* @__PURE__ */ y({
  __name: "MazBottomSheet",
  props: {
    noClose: { type: Boolean, default: !1 },
    noPadding: { type: Boolean, default: !1 }
  },
  emits: ["update:model-value", "open", "close"],
  setup(e, { emit: t }) {
    const n = V(() => import("./MazBtn-Ho_JsY-3.mjs")), l = t;
    return (s, a) => (c(), p(K, {
      "transition-name": "bottom-sheet-anim",
      "backdrop-class": "--bottom-sheet",
      onClose: a[0] || (a[0] = (o) => s.$emit("close", o)),
      onOpen: a[1] || (a[1] = (o) => s.$emit("open", o)),
      "onUpdate:modelValue": a[2] || (a[2] = (o) => l("update:model-value", o))
    }, {
      default: u(({ close: o }) => [
        d(
          "div",
          {
            class: f(["m-bottom-sheet__container", {
              "maz-py-6": !e.noPadding
            }])
          },
          [
            k(" Slot content "),
            E(s.$slots, "default", { close: o }, () => [
              J
            ], !0),
            e.noClose ? k("v-if", !0) : (c(), p(C(n), {
              key: 0,
              size: "sm",
              class: "m-bottom-sheet__close",
              color: "transparent",
              onClick: o
            }, {
              default: u(() => [
                B(C(G), { class: "maz-text-lg" })
              ]),
              _: 2
              /* DYNAMIC */
            }, 1032, ["onClick"]))
          ],
          2
          /* CLASS */
        )
      ]),
      _: 3
      /* FORWARDED */
    }));
  }
}), Y = /* @__PURE__ */ A(Q, [["__scopeId", "data-v-46b34248"]]);
export {
  Y as M,
  A as _
};
