// maz-ui/components/chunks/MazBtn-xF3JhNDM.mjs
import "../assets/MazBtn.css";
import { defineComponent as M, defineAsyncComponent as h, useAttrs as E, useSlots as L, onBeforeMount as R, computed as t, openBlock as o, createBlock as u, resolveDynamicComponent as w, normalizeClass as s, withCtx as A, createElementBlock as g, renderSlot as c, unref as d, createCommentVNode as a, createElementVNode as D } from "vue";
const N = /* @__PURE__ */ M({
  __name: "MazBtn",
  props: {
    variant: {
      type: String,
      default: "button",
      validator: (e) => ["button", "link"].includes(e)
    },
    size: {
      type: String,
      default: "md",
      validator: (e) => ["mini", "xs", "sm", "md", "lg", "xl"].includes(e)
    },
    color: {
      type: String,
      default: "primary"
    },
    type: {
      type: String,
      default: "button",
      validator: (e) => ["button", "submit", "reset"].includes(e)
    },
    rounded: { type: Boolean, default: !1 },
    noRounded: { type: Boolean, default: !1 },
    outline: { type: Boolean, default: !1 },
    pastel: { type: Boolean, default: !1 },
    block: { type: Boolean, default: !1 },
    noUnderline: { type: Boolean, default: !1 },
    noLeading: { type: Boolean, default: !1 },
    loading: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    fab: { type: Boolean, default: !1 },
    icon: { type: String, default: void 0 },
    leftIcon: { type: String, default: void 0 },
    rightIcon: { type: String, default: void 0 },
    noPadding: { type: Boolean, default: !1 },
    noElevation: { type: Boolean, default: !1 }
  },
  setup(e) {
    const f = h(() => import("./MazSpinner-wkIScBpQ.mjs")), l = h(() => import("./MazIcon-RrJDzX3V.mjs")), { href: v, to: m } = E(), z = L(), n = e;
    R(() => {
      n.icon && !n.fab && console.error('[maz-ui](MazBtn) the prop "icon" must be used only with "fab" props');
    });
    const b = t(() => v ? "a" : m ? "router-link" : "button"), k = t(
      () => n.pastel ? `--${n.color}-pastel` : n.outline ? `--${n.color}-outline` : `--${n.color}`
    ), y = t(
      () => (n.loading || n.disabled) && b.value === "button"
    ), I = t(() => y.value ? "--cursor-default" : "--cursor-pointer"), x = t(() => `--is-${n.variant}`), i = t(() => n.loading && n.variant === "button"), p = t(() => !!z["left-icon"] || n.leftIcon), B = t(() => !!z["right-icon"] || n.rightIcon), C = t(() => p.value || B.value), S = t(() => n.fab && n.icon), $ = t(() => b.value === "button" ? n.type : void 0);
    return (r, U) => (o(), u(w(b.value), {
      disabled: y.value,
      class: s(["m-btn", [
        `--${e.size}`,
        k.value,
        I.value,
        x.value,
        {
          "--block": e.block,
          "--no-underline": e.noUnderline,
          "--no-leading": e.noLeading,
          "--fab": e.fab,
          "--loading": e.loading,
          "--disabled": y.value,
          "--icon": C.value,
          "--rounded": e.rounded,
          "--no-rounded": e.noRounded,
          "--no-padding": e.noPadding,
          "--no-elevation": e.noElevation
        }
      ]]),
      type: $.value
    }, {
      default: A(() => [
        p.value ? (o(), g(
          "div",
          {
            key: 0,
            class: s(["m-btn__icon-left maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "left-icon", {}, () => [
              e.leftIcon ? (o(), u(d(l), {
                key: 0,
                name: e.leftIcon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        S.value ? (o(), g(
          "div",
          {
            key: 1,
            class: s(["m-btn__icon maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "icon", {}, () => [
              e.icon ? (o(), u(d(l), {
                key: 0,
                name: e.icon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        D(
          "span",
          {
            class: s(["maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "default", {}, void 0, !0)
          ],
          2
          /* CLASS */
        ),
        B.value ? (o(), g(
          "div",
          {
            key: 2,
            class: s(["m-btn__icon-right maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "right-icon", {}, () => [
              e.rightIcon ? (o(), u(d(l), {
                key: 0,
                name: e.rightIcon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        i.value ? (o(), u(d(f), {
          key: 3,
          size: "2em",
          color: e.color,
          class: "maz-absolute"
        }, null, 8, ["color"])) : a("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["disabled", "class", "type"]));
  }
}), P = (e, f) => {
  const l = e.__vccOpts || e;
  for (const [v, m] of f)
    l[v] = m;
  return l;
}, O = /* @__PURE__ */ P(N, [["__scopeId", "data-v-3f2db58e"]]);
export {
  O as M,
  P as _
};
