// maz-ui/components/chunks/MazBtn-6-KvWOG9.mjs
import "../assets/MazBtn.css";
import { defineComponent as M, defineAsyncComponent as z, useAttrs as E, useSlots as L, onBeforeMount as R, computed as t, openBlock as o, createBlock as i, resolveDynamicComponent as w, normalizeClass as u, withCtx as A, createElementBlock as v, renderSlot as r, unref as d, createCommentVNode as a, createElementVNode as D } from "vue";
import { _ as N } from "./MazCarousel-xYPtomV-.mjs";
const P = /* @__PURE__ */ M({
  __name: "MazBtn",
  props: {
    variant: {
      type: String,
      default: "button",
      validator: (e) => ["button", "link"].includes(e)
    },
    size: {
      type: String,
      default: "md",
      validator: (e) => ["mini", "xs", "sm", "md", "lg", "xl"].includes(e)
    },
    color: {
      type: String,
      default: "primary"
    },
    type: {
      type: String,
      default: "button",
      validator: (e) => ["button", "submit", "reset"].includes(e)
    },
    rounded: { type: Boolean, default: !1 },
    noRounded: { type: Boolean, default: !1 },
    outline: { type: Boolean, default: !1 },
    pastel: { type: Boolean, default: !1 },
    block: { type: Boolean, default: !1 },
    noUnderline: { type: Boolean, default: !1 },
    noLeading: { type: Boolean, default: !1 },
    loading: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    fab: { type: Boolean, default: !1 },
    icon: { type: String, default: void 0 },
    leftIcon: { type: String, default: void 0 },
    rightIcon: { type: String, default: void 0 },
    noPadding: { type: Boolean, default: !1 },
    noElevation: { type: Boolean, default: !1 }
  },
  setup(e) {
    const p = z(() => import("./MazSpinner-mXSRdmmW.mjs")), c = z(() => import("./MazIcon-RrJDzX3V.mjs")), { href: B, to: h } = E(), b = L(), n = e;
    R(() => {
      n.icon && !n.fab && console.error('[maz-ui](MazBtn) the prop "icon" must be used only with "fab" props');
    });
    const f = t(() => B ? "a" : h ? "router-link" : "button"), k = t(
      () => n.pastel ? `--${n.color}-pastel` : n.outline ? `--${n.color}-outline` : `--${n.color}`
    ), m = t(
      () => (n.loading || n.disabled) && f.value === "button"
    ), I = t(() => m.value ? "--cursor-default" : "--cursor-pointer"), x = t(() => `--is-${n.variant}`), l = t(() => n.loading && n.variant === "button"), y = t(() => !!b["left-icon"] || n.leftIcon), g = t(() => !!b["right-icon"] || n.rightIcon), C = t(() => y.value || g.value), S = t(() => n.fab && n.icon), $ = t(() => f.value === "button" ? n.type : void 0);
    return (s, U) => (o(), i(w(f.value), {
      disabled: m.value,
      class: u(["m-btn", [
        `--${e.size}`,
        k.value,
        I.value,
        x.value,
        {
          "--block": e.block,
          "--no-underline": e.noUnderline,
          "--no-leading": e.noLeading,
          "--fab": e.fab,
          "--loading": e.loading,
          "--disabled": m.value,
          "--icon": C.value,
          "--rounded": e.rounded,
          "--no-rounded": e.noRounded,
          "--no-padding": e.noPadding,
          "--no-elevation": e.noElevation
        }
      ]]),
      type: $.value
    }, {
      default: A(() => [
        y.value ? (o(), v(
          "div",
          {
            key: 0,
            class: u(["m-btn__icon-left maz-flex maz-flex-center", { "maz-invisible": l.value }])
          },
          [
            r(s.$slots, "left-icon", {}, () => [
              e.leftIcon ? (o(), i(d(c), {
                key: 0,
                name: e.leftIcon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        S.value ? (o(), v(
          "div",
          {
            key: 1,
            class: u(["m-btn__icon maz-flex maz-flex-center", { "maz-invisible": l.value }])
          },
          [
            r(s.$slots, "icon", {}, () => [
              e.icon ? (o(), i(d(c), {
                key: 0,
                name: e.icon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        D(
          "span",
          {
            class: u(["maz-flex maz-flex-center", { "maz-invisible": l.value }])
          },
          [
            r(s.$slots, "default", {}, void 0, !0)
          ],
          2
          /* CLASS */
        ),
        g.value ? (o(), v(
          "div",
          {
            key: 2,
            class: u(["m-btn__icon-right maz-flex maz-flex-center", { "maz-invisible": l.value }])
          },
          [
            r(s.$slots, "right-icon", {}, () => [
              e.rightIcon ? (o(), i(d(c), {
                key: 0,
                name: e.rightIcon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        l.value ? (o(), i(d(p), {
          key: 3,
          size: "2em",
          color: e.color,
          class: "maz-absolute"
        }, null, 8, ["color"])) : a("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["disabled", "class", "type"]));
  }
}), j = /* @__PURE__ */ N(P, [["__scopeId", "data-v-3f2db58e"]]);
export {
  j as default
};
