// maz-ui/components/chunks/MazBtn-1wfj_m47.mjs
import "../assets/MazBtn.css";
import { defineComponent as M, defineAsyncComponent as h, useAttrs as E, useSlots as L, onBeforeMount as O, computed as n, openBlock as o, createBlock as u, resolveDynamicComponent as P, normalizeClass as s, withCtx as R, createElementBlock as g, renderSlot as c, unref as d, createCommentVNode as a, createElementVNode as j } from "vue";
const w = /* @__PURE__ */ M({
  __name: "MazBtn",
  props: {
    variant: {
      type: String,
      default: "button",
      validator: (e) => ["button", "link"].includes(e)
    },
    size: {
      type: String,
      default: "md",
      validator: (e) => ["mini", "xs", "sm", "md", "lg", "xl"].includes(e)
    },
    color: {
      type: String,
      default: "primary"
    },
    type: {
      type: String,
      default: "button",
      validator: (e) => ["button", "submit", "reset"].includes(e)
    },
    rounded: { type: Boolean, default: !1 },
    noRounded: { type: Boolean, default: !1 },
    outline: { type: Boolean, default: !1 },
    pastel: { type: Boolean, default: !1 },
    block: { type: Boolean, default: !1 },
    noUnderline: { type: Boolean, default: !1 },
    noLeading: { type: Boolean, default: !1 },
    loading: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    fab: { type: Boolean, default: !1 },
    icon: { type: String, default: void 0 },
    leftIcon: { type: String, default: void 0 },
    rightIcon: { type: String, default: void 0 },
    noPadding: { type: Boolean, default: !1 },
    noElevation: { type: Boolean, default: !1 }
  },
  setup(e) {
    const f = h(() => import("./MazSpinner-tcjX1fe3.mjs")), l = h(() => import("./MazIcon-RrJDzX3V.mjs")), { href: v, to: m } = E(), z = L(), t = e;
    O(() => {
      t.icon && !t.fab && console.error('[maz-ui](MazBtn) the prop "icon" must be used only with "fab" props');
    });
    const b = n(() => v ? "a" : m ? "router-link" : "button"), k = n(
      () => t.pastel ? `--${t.color}-pastel` : t.outline ? `--${t.color}-outline` : `--${t.color}`
    ), y = n(
      () => (t.loading || t.disabled) && b.value === "button"
    ), I = n(() => y.value ? "--cursor-default" : "--cursor-pointer"), S = n(() => `--is-${t.variant}`), i = n(() => t.loading && t.variant === "button"), p = n(() => !!z["left-icon"] || t.leftIcon), B = n(() => !!z["right-icon"] || t.rightIcon), x = n(() => p.value || B.value), C = n(() => t.fab && t.icon), $ = n(() => b.value === "button" ? t.type : void 0);
    return (r, N) => (o(), u(P(b.value), {
      disabled: y.value,
      class: s(["m-btn", [
        `--${e.size}`,
        k.value,
        I.value,
        S.value,
        {
          "--block": e.block,
          "--no-underline": e.noUnderline,
          "--no-leading": e.noLeading,
          "--fab": e.fab,
          "--loading": e.loading,
          "--disabled": y.value,
          "--icon": x.value,
          "--rounded": e.rounded,
          "--no-rounded": e.noRounded,
          "--no-padding": e.noPadding,
          "--no-elevation": e.noElevation
        }
      ]]),
      type: $.value
    }, {
      default: R(() => [
        p.value ? (o(), g(
          "div",
          {
            key: 0,
            class: s(["m-btn__icon-left maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "left-icon", {}, () => [
              e.leftIcon ? (o(), u(d(l), {
                key: 0,
                name: e.leftIcon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        C.value ? (o(), g(
          "div",
          {
            key: 1,
            class: s(["m-btn__icon maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "icon", {}, () => [
              e.icon ? (o(), u(d(l), {
                key: 0,
                name: e.icon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        j(
          "span",
          {
            class: s(["maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "default", {}, void 0, !0)
          ],
          2
          /* CLASS */
        ),
        B.value ? (o(), g(
          "div",
          {
            key: 2,
            class: s(["m-btn__icon-right maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            c(r.$slots, "right-icon", {}, () => [
              e.rightIcon ? (o(), u(d(l), {
                key: 0,
                name: e.rightIcon
              }, null, 8, ["name"])) : a("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : a("v-if", !0),
        i.value ? (o(), u(d(f), {
          key: 3,
          size: "2em",
          color: e.color,
          class: "maz-absolute"
        }, null, 8, ["color"])) : a("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["disabled", "class", "type"]));
  }
}), A = (e, f) => {
  const l = e.__vccOpts || e;
  for (const [v, m] of f)
    l[v] = m;
  return l;
}, D = /* @__PURE__ */ A(w, [["__scopeId", "data-v-3f2db58e"]]), V = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: D
}, Symbol.toStringTag, { value: "Module" }));
export {
  V as M,
  A as _
};
