// maz-ui/components/chunks/MazCarousel-xYPtomV-.mjs
import "../assets/MazCarousel.css";
import { defineComponent as x, defineAsyncComponent as m, useSlots as L, ref as p, openBlock as c, createElementBlock as r, normalizeClass as v, renderSlot as a, createCommentVNode as n, createVNode as i, unref as u, withCtx as I, createElementVNode as y, pushScopeId as W, popScopeId as $ } from "vue";
const N = (t) => (W("data-v-e5ae2ebf"), t = t(), $(), t), T = { class: "m-carousel" }, V = { key: 0 }, w = {
  key: 1,
  class: "m-carousel__header__actions"
}, E = /* @__PURE__ */ N(() => /* @__PURE__ */ y(
  "div",
  { class: "m-carousel__items__spacer" },
  null,
  -1
  /* HOISTED */
)), A = /* @__PURE__ */ x({
  __name: "MazCarousel",
  props: {
    noScrollBtn: { type: Boolean, default: !1 }
  },
  setup(t) {
    const s = m(() => import("./MazBtn-6-KvWOG9.mjs")), l = m(() => import("./chevron-left-48sF6neb.mjs")), d = m(() => import("./chevron-right-VgbFQPBM.mjs")), _ = t, h = L(), S = p(!1), C = p(!1), f = p(), z = () => !_.noScrollBtn || h.title, k = () => !!h.title, B = () => {
      const e = f.value;
      e == null || e.scrollTo(e.scrollLeft + e.clientWidth, 0);
    }, M = () => {
      const e = f.value;
      e == null || e.scrollTo(e.scrollLeft - e.clientWidth, 0);
    }, b = (e) => {
      const o = e.target;
      if (!o)
        return;
      S.value = o.scrollLeft >= 20;
      const g = o.scrollWidth - o.clientWidth;
      C.value = o.scrollLeft >= g - 20;
    };
    return (e, o) => (c(), r("div", T, [
      z() ? (c(), r(
        "div",
        {
          key: 0,
          class: v(["m-carousel__header", { "--has-title": k() }])
        },
        [
          k() ? (c(), r("div", V, [
            a(e.$slots, "title", {}, void 0, !0)
          ])) : n("v-if", !0),
          t.noScrollBtn ? n("v-if", !0) : (c(), r("div", w, [
            i(u(s), {
              color: "transparent",
              class: v(["m-carousel__btn", { "--muted": !S.value }]),
              "no-elevation": "",
              fab: "",
              onClick: M
            }, {
              default: I(() => [
                a(e.$slots, "previous-icon", {}, () => [
                  i(u(l), { class: "maz-text-lg" })
                ], !0)
              ]),
              _: 3
              /* FORWARDED */
            }, 8, ["class"]),
            i(u(s), {
              color: "transparent",
              class: v(["m-carousel__btn", { "--muted": C.value }]),
              fab: "",
              "no-elevation": "",
              onClick: B
            }, {
              default: I(() => [
                a(e.$slots, "next-icon", {}, () => [
                  i(u(d), { class: "maz-text-lg" })
                ], !0)
              ]),
              _: 3
              /* FORWARDED */
            }, 8, ["class"])
          ]))
        ],
        2
        /* CLASS */
      )) : n("v-if", !0),
      y(
        "div",
        {
          ref_key: "MazCarouselItems",
          ref: f,
          class: "m-carousel__items",
          onScroll: b
        },
        [
          n(" Insert your items "),
          a(e.$slots, "default", {}, void 0, !0),
          E
        ],
        544
        /* NEED_HYDRATION, NEED_PATCH */
      )
    ]));
  }
}), H = (t, s) => {
  const l = t.__vccOpts || t;
  for (const [d, _] of s)
    l[d] = _;
  return l;
}, j = /* @__PURE__ */ H(A, [["__scopeId", "data-v-e5ae2ebf"]]);
export {
  j as M,
  H as _
};
