// maz-ui/components/chunks/MazCheckbox-pXHcef3K.mjs
import "../assets/MazCheckbox.css";
import { defineComponent as g, useCssVars as S, getCurrentInstance as C, computed as l, openBlock as I, createElementBlock as A, unref as c, normalizeClass as m, normalizeStyle as w, createElementVNode as f, mergeProps as B, createVNode as N, renderSlot as M } from "vue";
import { u as j, _ as E } from "./MazSelect-QrKXhl1v.mjs";
import O from "./check-_5Up9njX.mjs";
const _ = ["for", "aria-checked"], q = ["id", "checked", "disabled", "name"], D = /* @__PURE__ */ g({
  inheritAttrs: !1,
  __name: "MazCheckbox",
  props: {
    style: {
      type: [String, Array, Object],
      default: void 0
    },
    class: {
      type: [String, Array, Object],
      default: void 0
    },
    modelValue: {
      type: [Boolean, Array],
      default: void 0
    },
    id: { type: String, default: void 0 },
    color: {
      type: String,
      default: "primary"
    },
    value: { type: [String, Number, Boolean], default: void 0 },
    name: { type: String, default: "m-checkbox" },
    size: { type: String, default: "md" },
    disabled: { type: Boolean, default: !1 }
  },
  emits: [
    /* emitted when value change */
    "update:model-value",
    /* emited when value change */
    "change"
  ],
  setup(o, { emit: p }) {
    S((a) => ({
      "3a21591b": x.value,
      "4953c77d": h.value,
      d5250ce8: k.value,
      "105d84cf": z.value
    }));
    const b = C(), e = o, n = p, s = j({
      componentName: "MazCheckbox",
      instance: b,
      providedId: e.id
    }), d = l(
      () => typeof e.value != "boolean" && Array.isArray(e.modelValue) ? e.modelValue.includes(e.value) : typeof e.modelValue == "boolean" ? e.modelValue : !1
    ), h = l(() => {
      switch (e.size) {
        case "xl":
          return "2.25rem";
        case "lg":
          return "2rem";
        default:
          return "1.625rem";
        case "sm":
          return "1.425rem";
        case "xs":
          return "1.325rem";
        case "mini":
          return "1.2rem";
      }
    }), y = l(() => {
      switch (e.size) {
        case "xl":
          return "maz-text-2xl";
        case "lg":
          return "maz-text-xl";
        default:
          return "maz-text-lg";
        case "sm":
          return "maz-text-base";
        case "xs":
          return "maz-text-sm";
        case "mini":
          return "maz-text-xs";
      }
    }), x = l(() => `var(--maz-color-${e.color}-contrast)`), k = l(() => `var(--maz-color-${e.color})`), z = l(
      () => ["black", "transparent"].includes(e.color) ? "var(--maz-color-muted)" : `var(--maz-color-${e.color}-alpha)`
    );
    function V(a) {
      ["Space"].includes(a.code) && (a.preventDefault(), u(e.value ?? !e.modelValue));
    }
    function v(a) {
      return typeof a == "boolean" && (typeof e.modelValue == "boolean" || e.modelValue === void 0 || e.modelValue === null) ? !e.modelValue : Array.isArray(e.modelValue) && typeof a != "boolean" ? e.modelValue.includes(a) ? e.modelValue.filter((t) => t !== a) : [...e.modelValue, a] : [a];
    }
    function u(a) {
      const t = v(a);
      n("update:model-value", t), n("change", t);
    }
    return (a, t) => (I(), A("label", {
      for: c(s),
      class: m(["m-checkbox", [{ "--disabled": o.disabled }, e.class]]),
      tabindex: "0",
      style: w(o.style),
      role: "checkbox",
      "aria-checked": d.value,
      onKeydown: V
    }, [
      f("input", B({
        id: c(s),
        checked: d.value
      }, a.$attrs, {
        tabindex: "-1",
        disabled: o.disabled,
        name: o.name,
        type: "checkbox",
        onChange: t[0] || (t[0] = (r) => {
          var i;
          return u(o.value ?? ((i = r == null ? void 0 : r.target) == null ? void 0 : i.checked));
        })
      }), null, 16, q),
      f("span", null, [
        N(c(O), {
          class: m(["check-icon", y.value])
        }, null, 8, ["class"])
      ]),
      M(a.$slots, "default", {}, void 0, !0)
    ], 46, _));
  }
}), $ = /* @__PURE__ */ E(D, [["__scopeId", "data-v-05f6558f"]]);
export {
  $ as default
};
