// maz-ui/components/chunks/MazPickerYearSwitcher-QcFW_c-2.mjs
import "../assets/MazPickerYearSwitcher.css";
import { defineComponent as g, ref as b, computed as w, openBlock as m, createElementBlock as _, createElementVNode as u, createVNode as o, withModifiers as l, withCtx as i, unref as s, Fragment as D, renderList as S, createBlock as Y, normalizeClass as B, createTextVNode as M, toDisplayString as q } from "vue";
import { d as c, a as I, i as y, _ as N } from "./MazPicker-DBOTnZbk.mjs";
import d from "./MazBtn-v6ANLVfx.mjs";
import { X as V } from "./x-mark-d7yqulTI.mjs";
import { C as z } from "./MazPickerCalendar-flhTDFlr.mjs";
const E = { class: "maz-picker-year-switcher" }, L = { class: "maz-picker-year-switcher__header" }, P = { class: "maz-flex maz-space-x-2" }, T = { class: "maz-picker-year-switcher__main" }, X = /* @__PURE__ */ g({
  __name: "MazPickerYearSwitcher",
  props: {
    color: { type: String, required: !0 },
    locale: { type: String, required: !0 },
    calendarDate: { type: String, required: !0 }
  },
  emits: ["update:calendar-date", "close"],
  setup(n, { emit: k }) {
    const p = n, f = k, t = b(p.calendarDate), v = w(() => Array.from({ length: 15 }, (a, r) => r - 7).map((a) => {
      const r = c(t.value).get("year"), e = c(t.value).set("year", r + a);
      return {
        label: I(e.format(), p.locale, {
          year: "numeric"
        }),
        date: e
      };
    })), h = (a) => {
      f("update:calendar-date", c(a).format()), f("close");
    }, x = () => {
      t.value = c(t.value).subtract(7, "year").format();
    }, C = () => {
      t.value = c(t.value).add(7, "year").format();
    };
    return (a, r) => (m(), _("div", E, [
      u("div", L, [
        u("div", P, [
          o(d, {
            size: "xs",
            color: "transparent",
            type: "button",
            onClick: l(x, ["stop"])
          }, {
            default: i(() => [
              o(s(z), { class: "maz-text-lg" })
            ]),
            _: 1
            /* STABLE */
          }, 8, ["onClick"]),
          o(d, {
            size: "xs",
            color: "transparent",
            type: "button",
            onClick: l(C, ["stop"])
          }, {
            default: i(() => [
              o(s(z), { class: "maz-rotate-180 maz-text-lg" })
            ]),
            _: 1
            /* STABLE */
          }, 8, ["onClick"])
        ]),
        o(d, {
          size: "xs",
          color: "transparent",
          type: "button",
          onClick: r[0] || (r[0] = l((e) => a.$emit("close", e), ["stop"]))
        }, {
          default: i(() => [
            o(s(V), { class: "maz-text-lg" })
          ]),
          _: 1
          /* STABLE */
        })
      ]),
      u("div", T, [
        (m(!0), _(
          D,
          null,
          S(v.value, (e) => (m(), Y(d, {
            key: e.label,
            size: "sm",
            type: "button",
            class: B({
              "--is-selected": s(y)(e.date, n.calendarDate, "year")
            }),
            color: s(y)(e.date, n.calendarDate, "year") ? n.color : "transparent",
            onClick: l(($) => h(e.date), ["stop"])
          }, {
            default: i(() => [
              M(
                q(e.label),
                1
                /* TEXT */
              )
            ]),
            _: 2
            /* DYNAMIC */
          }, 1032, ["class", "color", "onClick"]))),
          128
          /* KEYED_FRAGMENT */
        ))
      ])
    ]));
  }
}), K = /* @__PURE__ */ N(X, [["__scopeId", "data-v-698468b9"]]);
export {
  K as default
};
