// maz-ui/components/chunks/MazPicker-DBOTnZbk.mjs
import "../assets/MazPicker.css";
import { nextTick as be, defineComponent as Me, defineAsyncComponent as te, computed as R, openBlock as _, createElementBlock as we, normalizeClass as ge, createBlock as re, unref as ee, createCommentVNode as oe, createElementVNode as $e, getCurrentInstance as Ve, ref as K, onBeforeMount as ze, onMounted as Be, onUnmounted as Ee, watch as de, withDirectives as De, normalizeStyle as Ae, mergeProps as He, withCtx as pe, createVNode as fe, Transition as Pe, vShow as je } from "vue";
var se = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function ue(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var Ye = { exports: {} };
(function(e, u) {
  (function(h, y) {
    e.exports = y();
  })(se, function() {
    var h = 1e3, y = 6e4, $ = 36e5, t = "millisecond", k = "second", M = "minute", T = "hour", b = "day", x = "week", S = "month", W = "quarter", J = "year", Q = "date", v = "Invalid Date", m = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, L = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, O = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(c) {
      var i = ["th", "st", "nd", "rd"], n = c % 100;
      return "[" + c + (i[(n - 20) % 10] || i[n] || i[0]) + "]";
    } }, P = function(c, i, n) {
      var s = String(c);
      return !s || s.length >= i ? c : "" + Array(i + 1 - s.length).join(n) + c;
    }, A = { s: P, z: function(c) {
      var i = -c.utcOffset(), n = Math.abs(i), s = Math.floor(n / 60), r = n % 60;
      return (i <= 0 ? "+" : "-") + P(s, 2, "0") + ":" + P(r, 2, "0");
    }, m: function c(i, n) {
      if (i.date() < n.date())
        return -c(n, i);
      var s = 12 * (n.year() - i.year()) + (n.month() - i.month()), r = i.clone().add(s, S), l = n - r < 0, d = i.clone().add(s + (l ? -1 : 1), S);
      return +(-(s + (n - r) / (l ? r - d : d - r)) || 0);
    }, a: function(c) {
      return c < 0 ? Math.ceil(c) || 0 : Math.floor(c);
    }, p: function(c) {
      return { M: S, y: J, w: x, d: b, D: Q, h: T, m: M, s: k, ms: t, Q: W }[c] || String(c || "").toLowerCase().replace(/s$/, "");
    }, u: function(c) {
      return c === void 0;
    } }, q = "en", C = {};
    C[q] = O;
    var H = "$isDayjsObject", z = function(c) {
      return c instanceof E || !(!c || !c[H]);
    }, B = function c(i, n, s) {
      var r;
      if (!i)
        return q;
      if (typeof i == "string") {
        var l = i.toLowerCase();
        C[l] && (r = l), n && (C[l] = n, r = l);
        var d = i.split("-");
        if (!r && d.length > 1)
          return c(d[0]);
      } else {
        var g = i.name;
        C[g] = i, r = g;
      }
      return !s && r && (q = r), r || !s && q;
    }, Y = function(c, i) {
      if (z(c))
        return c.clone();
      var n = typeof i == "object" ? i : {};
      return n.date = c, n.args = arguments, new E(n);
    }, D = A;
    D.l = B, D.i = z, D.w = function(c, i) {
      return Y(c, { locale: i.$L, utc: i.$u, x: i.$x, $offset: i.$offset });
    };
    var E = function() {
      function c(n) {
        this.$L = B(n.locale, null, !0), this.parse(n), this.$x = this.$x || n.x || {}, this[H] = !0;
      }
      var i = c.prototype;
      return i.parse = function(n) {
        this.$d = function(s) {
          var r = s.date, l = s.utc;
          if (r === null)
            return /* @__PURE__ */ new Date(NaN);
          if (D.u(r))
            return /* @__PURE__ */ new Date();
          if (r instanceof Date)
            return new Date(r);
          if (typeof r == "string" && !/Z$/i.test(r)) {
            var d = r.match(m);
            if (d) {
              var g = d[2] - 1 || 0, o = (d[7] || "0").substring(0, 3);
              return l ? new Date(Date.UTC(d[1], g, d[3] || 1, d[4] || 0, d[5] || 0, d[6] || 0, o)) : new Date(d[1], g, d[3] || 1, d[4] || 0, d[5] || 0, d[6] || 0, o);
            }
          }
          return new Date(r);
        }(n), this.init();
      }, i.init = function() {
        var n = this.$d;
        this.$y = n.getFullYear(), this.$M = n.getMonth(), this.$D = n.getDate(), this.$W = n.getDay(), this.$H = n.getHours(), this.$m = n.getMinutes(), this.$s = n.getSeconds(), this.$ms = n.getMilliseconds();
      }, i.$utils = function() {
        return D;
      }, i.isValid = function() {
        return this.$d.toString() !== v;
      }, i.isSame = function(n, s) {
        var r = Y(n);
        return this.startOf(s) <= r && r <= this.endOf(s);
      }, i.isAfter = function(n, s) {
        return Y(n) < this.startOf(s);
      }, i.isBefore = function(n, s) {
        return this.endOf(s) < Y(n);
      }, i.$g = function(n, s, r) {
        return D.u(n) ? this[s] : this.set(r, n);
      }, i.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, i.valueOf = function() {
        return this.$d.getTime();
      }, i.startOf = function(n, s) {
        var r = this, l = !!D.u(s) || s, d = D.p(n), g = function(I, V) {
          var U = D.w(r.$u ? Date.UTC(r.$y, V, I) : new Date(r.$y, V, I), r);
          return l ? U : U.endOf(b);
        }, o = function(I, V) {
          return D.w(r.toDate()[I].apply(r.toDate("s"), (l ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(V)), r);
        }, a = this.$W, f = this.$M, w = this.$D, F = "set" + (this.$u ? "UTC" : "");
        switch (d) {
          case J:
            return l ? g(1, 0) : g(31, 11);
          case S:
            return l ? g(1, f) : g(0, f + 1);
          case x:
            var N = this.$locale().weekStart || 0, Z = (a < N ? a + 7 : a) - N;
            return g(l ? w - Z : w + (6 - Z), f);
          case b:
          case Q:
            return o(F + "Hours", 0);
          case T:
            return o(F + "Minutes", 1);
          case M:
            return o(F + "Seconds", 2);
          case k:
            return o(F + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, i.endOf = function(n) {
        return this.startOf(n, !1);
      }, i.$set = function(n, s) {
        var r, l = D.p(n), d = "set" + (this.$u ? "UTC" : ""), g = (r = {}, r[b] = d + "Date", r[Q] = d + "Date", r[S] = d + "Month", r[J] = d + "FullYear", r[T] = d + "Hours", r[M] = d + "Minutes", r[k] = d + "Seconds", r[t] = d + "Milliseconds", r)[l], o = l === b ? this.$D + (s - this.$W) : s;
        if (l === S || l === J) {
          var a = this.clone().set(Q, 1);
          a.$d[g](o), a.init(), this.$d = a.set(Q, Math.min(this.$D, a.daysInMonth())).$d;
        } else
          g && this.$d[g](o);
        return this.init(), this;
      }, i.set = function(n, s) {
        return this.clone().$set(n, s);
      }, i.get = function(n) {
        return this[D.p(n)]();
      }, i.add = function(n, s) {
        var r, l = this;
        n = Number(n);
        var d = D.p(s), g = function(f) {
          var w = Y(l);
          return D.w(w.date(w.date() + Math.round(f * n)), l);
        };
        if (d === S)
          return this.set(S, this.$M + n);
        if (d === J)
          return this.set(J, this.$y + n);
        if (d === b)
          return g(1);
        if (d === x)
          return g(7);
        var o = (r = {}, r[M] = y, r[T] = $, r[k] = h, r)[d] || 1, a = this.$d.getTime() + n * o;
        return D.w(a, this);
      }, i.subtract = function(n, s) {
        return this.add(-1 * n, s);
      }, i.format = function(n) {
        var s = this, r = this.$locale();
        if (!this.isValid())
          return r.invalidDate || v;
        var l = n || "YYYY-MM-DDTHH:mm:ssZ", d = D.z(this), g = this.$H, o = this.$m, a = this.$M, f = r.weekdays, w = r.months, F = r.meridiem, N = function(V, U, X, G) {
          return V && (V[U] || V(s, l)) || X[U].slice(0, G);
        }, Z = function(V) {
          return D.s(g % 12 || 12, V, "0");
        }, I = F || function(V, U, X) {
          var G = V < 12 ? "AM" : "PM";
          return X ? G.toLowerCase() : G;
        };
        return l.replace(L, function(V, U) {
          return U || function(X) {
            switch (X) {
              case "YY":
                return String(s.$y).slice(-2);
              case "YYYY":
                return D.s(s.$y, 4, "0");
              case "M":
                return a + 1;
              case "MM":
                return D.s(a + 1, 2, "0");
              case "MMM":
                return N(r.monthsShort, a, w, 3);
              case "MMMM":
                return N(w, a);
              case "D":
                return s.$D;
              case "DD":
                return D.s(s.$D, 2, "0");
              case "d":
                return String(s.$W);
              case "dd":
                return N(r.weekdaysMin, s.$W, f, 2);
              case "ddd":
                return N(r.weekdaysShort, s.$W, f, 3);
              case "dddd":
                return f[s.$W];
              case "H":
                return String(g);
              case "HH":
                return D.s(g, 2, "0");
              case "h":
                return Z(1);
              case "hh":
                return Z(2);
              case "a":
                return I(g, o, !0);
              case "A":
                return I(g, o, !1);
              case "m":
                return String(o);
              case "mm":
                return D.s(o, 2, "0");
              case "s":
                return String(s.$s);
              case "ss":
                return D.s(s.$s, 2, "0");
              case "SSS":
                return D.s(s.$ms, 3, "0");
              case "Z":
                return d;
            }
            return null;
          }(V) || d.replace(":", "");
        });
      }, i.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, i.diff = function(n, s, r) {
        var l, d = this, g = D.p(s), o = Y(n), a = (o.utcOffset() - this.utcOffset()) * y, f = this - o, w = function() {
          return D.m(d, o);
        };
        switch (g) {
          case J:
            l = w() / 12;
            break;
          case S:
            l = w();
            break;
          case W:
            l = w() / 3;
            break;
          case x:
            l = (f - a) / 6048e5;
            break;
          case b:
            l = (f - a) / 864e5;
            break;
          case T:
            l = f / $;
            break;
          case M:
            l = f / y;
            break;
          case k:
            l = f / h;
            break;
          default:
            l = f;
        }
        return r ? l : D.a(l);
      }, i.daysInMonth = function() {
        return this.endOf(S).$D;
      }, i.$locale = function() {
        return C[this.$L];
      }, i.locale = function(n, s) {
        if (!n)
          return this.$L;
        var r = this.clone(), l = B(n, s, !0);
        return l && (r.$L = l), r;
      }, i.clone = function() {
        return D.w(this.$d, this);
      }, i.toDate = function() {
        return new Date(this.valueOf());
      }, i.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, i.toISOString = function() {
        return this.$d.toISOString();
      }, i.toString = function() {
        return this.$d.toUTCString();
      }, c;
    }(), j = E.prototype;
    return Y.prototype = j, [["$ms", t], ["$s", k], ["$m", M], ["$H", T], ["$W", b], ["$M", S], ["$y", J], ["$D", Q]].forEach(function(c) {
      j[c[1]] = function(i) {
        return this.$g(i, c[0], c[1]);
      };
    }), Y.extend = function(c, i) {
      return c.$i || (c(i, E, Y), c.$i = !0), Y;
    }, Y.locale = B, Y.isDayjs = z, Y.unix = function(c) {
      return Y(1e3 * c);
    }, Y.en = C[q], Y.Ls = C, Y.p = {}, Y;
  });
})(Ye);
var Ie = Ye.exports;
const p = /* @__PURE__ */ ue(Ie);
var ke = { exports: {} };
(function(e, u) {
  (function(h, y) {
    e.exports = y();
  })(se, function() {
    var h = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" }, y = /(\[[^[]*\])|([-_:/.,()\s]+)|(A|a|YYYY|YY?|MM?M?M?|Do|DD?|hh?|HH?|mm?|ss?|S{1,3}|z|ZZ?)/g, $ = /\d\d/, t = /\d\d?/, k = /\d*[^-_:/,()\s\d]+/, M = {}, T = function(v) {
      return (v = +v) + (v > 68 ? 1900 : 2e3);
    }, b = function(v) {
      return function(m) {
        this[v] = +m;
      };
    }, x = [/[+-]\d\d:?(\d\d)?|Z/, function(v) {
      (this.zone || (this.zone = {})).offset = function(m) {
        if (!m || m === "Z")
          return 0;
        var L = m.match(/([+-]|\d\d)/g), O = 60 * L[1] + (+L[2] || 0);
        return O === 0 ? 0 : L[0] === "+" ? -O : O;
      }(v);
    }], S = function(v) {
      var m = M[v];
      return m && (m.indexOf ? m : m.s.concat(m.f));
    }, W = function(v, m) {
      var L, O = M.meridiem;
      if (O) {
        for (var P = 1; P <= 24; P += 1)
          if (v.indexOf(O(P, 0, m)) > -1) {
            L = P > 12;
            break;
          }
      } else
        L = v === (m ? "pm" : "PM");
      return L;
    }, J = { A: [k, function(v) {
      this.afternoon = W(v, !1);
    }], a: [k, function(v) {
      this.afternoon = W(v, !0);
    }], S: [/\d/, function(v) {
      this.milliseconds = 100 * +v;
    }], SS: [$, function(v) {
      this.milliseconds = 10 * +v;
    }], SSS: [/\d{3}/, function(v) {
      this.milliseconds = +v;
    }], s: [t, b("seconds")], ss: [t, b("seconds")], m: [t, b("minutes")], mm: [t, b("minutes")], H: [t, b("hours")], h: [t, b("hours")], HH: [t, b("hours")], hh: [t, b("hours")], D: [t, b("day")], DD: [$, b("day")], Do: [k, function(v) {
      var m = M.ordinal, L = v.match(/\d+/);
      if (this.day = L[0], m)
        for (var O = 1; O <= 31; O += 1)
          m(O).replace(/\[|\]/g, "") === v && (this.day = O);
    }], M: [t, b("month")], MM: [$, b("month")], MMM: [k, function(v) {
      var m = S("months"), L = (S("monthsShort") || m.map(function(O) {
        return O.slice(0, 3);
      })).indexOf(v) + 1;
      if (L < 1)
        throw new Error();
      this.month = L % 12 || L;
    }], MMMM: [k, function(v) {
      var m = S("months").indexOf(v) + 1;
      if (m < 1)
        throw new Error();
      this.month = m % 12 || m;
    }], Y: [/[+-]?\d+/, b("year")], YY: [$, function(v) {
      this.year = T(v);
    }], YYYY: [/\d{4}/, b("year")], Z: x, ZZ: x };
    function Q(v) {
      var m, L;
      m = v, L = M && M.formats;
      for (var O = (v = m.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(B, Y, D) {
        var E = D && D.toUpperCase();
        return Y || L[D] || h[D] || L[E].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(j, c, i) {
          return c || i.slice(1);
        });
      })).match(y), P = O.length, A = 0; A < P; A += 1) {
        var q = O[A], C = J[q], H = C && C[0], z = C && C[1];
        O[A] = z ? { regex: H, parser: z } : q.replace(/^\[|\]$/g, "");
      }
      return function(B) {
        for (var Y = {}, D = 0, E = 0; D < P; D += 1) {
          var j = O[D];
          if (typeof j == "string")
            E += j.length;
          else {
            var c = j.regex, i = j.parser, n = B.slice(E), s = c.exec(n)[0];
            i.call(Y, s), B = B.replace(s, "");
          }
        }
        return function(r) {
          var l = r.afternoon;
          if (l !== void 0) {
            var d = r.hours;
            l ? d < 12 && (r.hours += 12) : d === 12 && (r.hours = 0), delete r.afternoon;
          }
        }(Y), Y;
      };
    }
    return function(v, m, L) {
      L.p.customParseFormat = !0, v && v.parseTwoDigitYear && (T = v.parseTwoDigitYear);
      var O = m.prototype, P = O.parse;
      O.parse = function(A) {
        var q = A.date, C = A.utc, H = A.args;
        this.$u = C;
        var z = H[1];
        if (typeof z == "string") {
          var B = H[2] === !0, Y = H[3] === !0, D = B || Y, E = H[2];
          Y && (E = H[2]), M = this.$locale(), !B && E && (M = L.Ls[E]), this.$d = function(n, s, r) {
            try {
              if (["x", "X"].indexOf(s) > -1)
                return new Date((s === "X" ? 1e3 : 1) * n);
              var l = Q(s)(n), d = l.year, g = l.month, o = l.day, a = l.hours, f = l.minutes, w = l.seconds, F = l.milliseconds, N = l.zone, Z = /* @__PURE__ */ new Date(), I = o || (d || g ? 1 : Z.getDate()), V = d || Z.getFullYear(), U = 0;
              d && !g || (U = g > 0 ? g - 1 : Z.getMonth());
              var X = a || 0, G = f || 0, le = w || 0, ce = F || 0;
              return N ? new Date(Date.UTC(V, U, I, X, G, le, ce + 60 * N.offset * 1e3)) : r ? new Date(Date.UTC(V, U, I, X, G, le, ce)) : new Date(V, U, I, X, G, le, ce);
            } catch {
              return /* @__PURE__ */ new Date("");
            }
          }(q, z, C), this.init(), E && E !== !0 && (this.$L = this.locale(E).$L), D && q != this.format(z) && (this.$d = /* @__PURE__ */ new Date("")), M = {};
        } else if (z instanceof Array)
          for (var j = z.length, c = 1; c <= j; c += 1) {
            H[1] = z[c - 1];
            var i = L.apply(this, H);
            if (i.isValid()) {
              this.$d = i.$d, this.$L = i.$L, this.init();
              break;
            }
            c === j && (this.$d = /* @__PURE__ */ new Date(""));
          }
        else
          P.call(this, A);
      };
    };
  });
})(ke);
var Ue = ke.exports;
const We = /* @__PURE__ */ ue(Ue);
var Se = { exports: {} };
(function(e, u) {
  (function(h, y) {
    e.exports = y();
  })(se, function() {
    return function(h, y, $) {
      y.prototype.isBetween = function(t, k, M, T) {
        var b = $(t), x = $(k), S = (T = T || "()")[0] === "(", W = T[1] === ")";
        return (S ? this.isAfter(b, M) : !this.isBefore(b, M)) && (W ? this.isBefore(x, M) : !this.isAfter(x, M)) || (S ? this.isBefore(b, M) : !this.isAfter(b, M)) && (W ? this.isAfter(x, M) : !this.isBefore(x, M));
      };
    };
  });
})(Se);
var qe = Se.exports;
const Fe = /* @__PURE__ */ ue(qe), ie = "__vue_click_away__", Oe = () => document.ontouchstart === null ? "touchstart" : "click", xe = async (e, u) => {
  Te(e);
  const h = u.instance, y = u.value, $ = typeof y == "function";
  if (!$)
    throw new Error("[maz-ui](vClickOutside) the callback should be a function");
  await be(), e[ie] = (k) => {
    if ((!e || !e.contains(k.target)) && y && $)
      return y.call(h, k);
  };
  const t = Oe();
  document.addEventListener(t, e[ie], !1);
}, Te = (e) => {
  const u = Oe();
  document.removeEventListener(u, e[ie], !1), delete e[ie];
}, Ne = (e, u) => {
  u.value !== u.oldValue && xe(e, u);
}, Ze = {
  mounted: xe,
  updated: Ne,
  unmounted: Te
}, Re = {
  month: "short",
  day: "numeric",
  year: "numeric"
}, ve = (e, u, h) => {
  if (u === void 0)
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute is required.");
  if (typeof u != "string")
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute must be a string.");
  const y = h ?? Re;
  try {
    const $ = e instanceof Date ? e : new Date(e);
    return new Intl.DateTimeFormat(u, y).format($);
  } catch ($) {
    throw new Error(`[maz-ui](FilterDate) ${$}`);
  }
};
var Le = { exports: {} };
(function(e, u) {
  (function(h, y) {
    e.exports = y();
  })(se, function() {
    return function(h, y) {
      y.prototype.weekday = function($) {
        var t = this.$locale().weekStart || 0, k = this.$W, M = (k < t ? k + 7 : k) - t;
        return this.$utils().u($) ? M : this.subtract(M, "day").add($, "day");
      };
    };
  });
})(Le);
var Je = Le.exports;
const Qe = /* @__PURE__ */ ue(Je), Xe = (e) => e ? (e = e.toString(), e.charAt(0).toUpperCase() + e.slice(1)) : "";
p.extend(Qe);
function he({
  value: e,
  locale: u,
  options: h
}) {
  return e ? Xe(ve(e, u, h)) : void 0;
}
function Ge({
  value: e,
  locale: u,
  options: h
}) {
  const y = he({
    value: e.start,
    locale: u,
    options: h
  }), $ = he({
    value: e.end,
    locale: u,
    options: h
  });
  return y || $ ? `${y || "..."} - ${$ || "..."}` : void 0;
}
function ct(e) {
  return p(e).startOf("month").day();
}
function dt(e, u) {
  const h = p().locale(e), y = [];
  for (let $ = 0; $ < 7; $++) {
    const t = h.day($ + u).toDate(), k = ve(t, e, { weekday: "short" });
    y.push(k);
  }
  return y;
}
function ft(e) {
  return p(e).daysInMonth();
}
function mt(e) {
  return p().isSame(e, "date");
}
function ht(e, u, h) {
  return p(e).isSame(u, h);
}
function Ke(e, u) {
  return p(e).day() === u;
}
function ye(e, u = "YYYY-MM-DD") {
  if (e)
    return p(e).format(u);
}
function _e(e, u = "YYYY-MM-DD") {
  return {
    start: ye(e.start, u),
    end: ye(e.end, u)
  };
}
function me({
  value: e,
  minDate: u,
  maxDate: h,
  format: y
}) {
  return u && p(e).isBefore(u) ? {
    newValue: u,
    newCurrentDate: p(u, y).format()
  } : h && p(e).isAfter(h) ? {
    newValue: h,
    newCurrentDate: p(h, y).format()
  } : {
    newValue: void 0,
    newCurrentDate: void 0
  };
}
function ae({
  value: e,
  disabledWeekly: u
}) {
  return u.some((h) => Ke(e, h));
}
function ne({
  value: e,
  disabledDates: u
}) {
  return u.some((h) => p(e).isSame(p(h), "date"));
}
const yt = function(e, u, h = 0, y = !0) {
  e.scrollTo({
    top: u.offsetTop - h,
    behavior: y ? "smooth" : "auto"
  });
}, vt = (e, u) => e.reduce((h, y) => Math.abs(y - u) < Math.abs(h - u) ? y : h);
function et() {
  try {
    return typeof window > "u" ? void 0 : window.navigator.language;
  } catch (e) {
    throw new Error(`[MazPhoneNumberInput] (browserLocale) ${e}`);
  }
}
async function tt() {
  try {
    const e = await fetch("https://ipwho.is"), { country_code: u } = await e.json();
    return u;
  } catch (e) {
    console.error(`[maz-ui](MazPicker)(fetchCountryCode) ${e}`);
  }
}
const at = { class: "m-picker-container__wrapper" }, nt = /* @__PURE__ */ Me({
  __name: "MazPickerContainer",
  props: {
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    calendarDate: { type: String, required: !0 },
    color: { type: String, required: !0 },
    locale: { type: String, required: !0 },
    noHeader: { type: Boolean, default: !1 },
    firstDayOfWeek: { type: Number, required: !0 },
    double: { type: Boolean, required: !0 },
    hasDate: { type: Boolean, required: !0 },
    minDate: { type: String, default: void 0 },
    maxDate: { type: String, default: void 0 },
    inline: { type: Boolean, required: !0 },
    noShortcuts: { type: Boolean, required: !0 },
    shortcuts: {
      type: Array,
      required: !0
    },
    shortcut: { type: String, default: void 0 },
    hasTime: { type: Boolean, required: !0 },
    isOpen: { type: Boolean, required: !0 },
    format: { type: String, required: !0 },
    isHour12: { type: Boolean, required: !0 },
    formatterOptions: {
      type: Object,
      required: !0
    },
    minuteInterval: { type: Number, required: !0 },
    disabled: { type: Boolean, required: !0 },
    disabledWeekly: { type: Array, required: !0 },
    disabledHours: { type: Array, required: !0 },
    disabledDates: { type: Array, required: !0 }
  },
  emits: ["update:model-value", "update:calendar-date", "close"],
  setup(e, { emit: u }) {
    const h = te(() => import("./MazPickerTime-BnE9Z2o_.mjs")), y = te(() => import("./MazPickerHeader-s-Vi7Luu.mjs")), $ = te(() => import("./MazPickerCalendar-flhTDFlr.mjs").then((b) => b.M)), t = e, k = u, M = R({
      get: () => t.modelValue,
      set: (b) => {
        k("update:model-value", b);
      }
    }), T = R({
      get: () => t.calendarDate,
      set: (b) => k("update:calendar-date", b)
    });
    return (b, x) => (_(), we(
      "div",
      {
        class: ge(["m-picker-container", {
          "--has-double": e.double,
          "--is-inline": e.inline,
          "--has-date": e.hasDate
        }])
      },
      [
        e.noHeader ? oe("v-if", !0) : (_(), re(ee(y), {
          key: 0,
          color: e.color,
          "has-time": e.hasTime,
          "model-value": e.modelValue,
          locale: e.locale,
          "calendar-date": T.value,
          "has-date": e.hasDate,
          "formatter-options": e.formatterOptions,
          double: e.double,
          "no-shortcuts": e.noShortcuts,
          class: "m-picker-container__header"
        }, null, 8, ["color", "has-time", "model-value", "locale", "calendar-date", "has-date", "formatter-options", "double", "no-shortcuts"])),
        $e("div", at, [
          e.hasDate ? (_(), re(ee($), {
            key: 0,
            modelValue: M.value,
            "onUpdate:modelValue": x[0] || (x[0] = (S) => M.value = S),
            "calendar-date": T.value,
            "onUpdate:calendarDate": x[1] || (x[1] = (S) => T.value = S),
            color: e.color,
            locale: e.locale,
            "has-time": e.hasTime,
            double: e.double,
            "min-date": e.minDate,
            "max-date": e.maxDate,
            "first-day-of-week": e.firstDayOfWeek,
            disabled: e.disabled,
            "disabled-weekly": e.disabledWeekly,
            "disabled-dates": e.disabledDates,
            shortcuts: e.shortcuts,
            shortcut: e.shortcut,
            "no-shortcuts": e.noShortcuts,
            class: "m-picker-container__calendar"
          }, null, 8, ["modelValue", "calendar-date", "color", "locale", "has-time", "double", "min-date", "max-date", "first-day-of-week", "disabled", "disabled-weekly", "disabled-dates", "shortcuts", "shortcut", "no-shortcuts"])) : oe("v-if", !0),
          e.hasTime ? (_(), re(ee(h), {
            key: 1,
            modelValue: M.value,
            "onUpdate:modelValue": x[2] || (x[2] = (S) => M.value = S),
            "calendar-date": T.value,
            "onUpdate:calendarDate": x[3] || (x[3] = (S) => T.value = S),
            "is-open": e.isOpen,
            color: e.color,
            locale: e.locale,
            "min-date": e.minDate,
            "max-date": e.maxDate,
            "has-date": e.hasDate,
            format: e.format,
            disabled: e.disabled,
            "disabled-hours": e.disabledHours,
            "minute-interval": e.minuteInterval,
            "formatter-options": e.formatterOptions,
            "is-hour12": e.isHour12,
            class: "m-picker-container__time"
          }, null, 8, ["modelValue", "calendar-date", "is-open", "color", "locale", "min-date", "max-date", "has-date", "format", "disabled", "disabled-hours", "minute-interval", "formatter-options", "is-hour12"])) : oe("v-if", !0)
        ])
      ],
      2
      /* CLASS */
    ));
  }
}), Ce = (e, u) => {
  const h = e.__vccOpts || e;
  for (const [y, $] of u)
    h[y] = $;
  return h;
}, rt = /* @__PURE__ */ Ce(nt, [["__scopeId", "data-v-eea709c5"]]), ot = {
  type: "button",
  tabindex: "-1",
  class: "m-picker__button"
}, it = /* @__PURE__ */ Me({
  inheritAttrs: !1,
  __name: "MazPicker",
  props: {
    style: {
      type: [String, Array, Object],
      default: void 0
    },
    class: {
      type: [String, Array, Object],
      default: void 0
    },
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    format: { type: String, default: "YYYY-MM-DD" },
    open: { type: Boolean, default: !1 },
    label: { type: String, default: void 0 },
    placeholder: { type: String, default: void 0 },
    inputDateStyle: {
      type: Object,
      default: () => ({
        dateStyle: "full"
      })
    },
    locale: { type: String, default: void 0 },
    noHeader: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    firstDayOfWeek: {
      type: Number,
      default: 0,
      validator: (e) => {
        const u = Array.from({ length: 7 }, (h, y) => y).includes(e);
        return u || console.error('[maz-ui](MazPicker) "first-day-of-week" should be between 0 and 6'), u;
      }
    },
    autoClose: { type: Boolean, default: !1 },
    customElementSelector: { type: String, default: void 0 },
    double: { type: Boolean, default: !1 },
    inline: { type: Boolean, default: !1 },
    color: {
      type: String,
      default: "primary"
    },
    pickerPosition: {
      type: String,
      default: void 0,
      validator: (e) => [
        "top",
        "top right",
        "top left",
        "bottom",
        "bottom right",
        "bottom left",
        "left",
        "right"
      ].includes(e)
    },
    time: { type: Boolean, default: !1 },
    onlyTime: { type: Boolean, default: !1 },
    minuteInterval: { type: Number, default: 5 },
    noUseBrowserLocale: { type: Boolean, default: !1 },
    noFetchLocal: { type: Boolean, default: !1 },
    noShortcuts: { type: Boolean, default: !1 },
    shortcuts: {
      type: Array,
      default: () => [
        {
          label: "Last 7 days",
          identifier: "last7Days",
          value: {
            start: p().subtract(6, "day").format("YYYY-MM-DD"),
            end: p().format("YYYY-MM-DD")
          }
        },
        {
          label: "Last 30 days",
          identifier: "last30Days",
          value: {
            start: p().subtract(29, "day").format("YYYY-MM-DD"),
            end: p().format("YYYY-MM-DD")
          }
        },
        {
          label: "This week",
          identifier: "thisWeek",
          value: {
            start: p().startOf("week").format("YYYY-MM-DD"),
            end: p().endOf("week").format("YYYY-MM-DD")
          }
        },
        {
          label: "Last week",
          identifier: "lastWeek",
          value: {
            start: p().subtract(1, "week").startOf("week").format("YYYY-MM-DD"),
            end: p().subtract(1, "week").endOf("week").format("YYYY-MM-DD")
          }
        },
        {
          label: "This month",
          identifier: "thisMonth",
          value: {
            start: p().set("date", 1).format("YYYY-MM-DD"),
            end: p().set("date", p().daysInMonth()).format("YYYY-MM-DD")
          }
        },
        {
          label: "This year",
          identifier: "thisYear",
          value: {
            start: p().startOf("year").format("YYYY-MM-DD"),
            end: p().endOf("year").format("YYYY-MM-DD")
          }
        },
        {
          label: "Last year",
          identifier: "lastYear",
          value: {
            start: p().subtract(1, "year").startOf("year").format("YYYY-MM-DD"),
            end: p().subtract(1, "year").endOf("year").format("YYYY-MM-DD")
          }
        }
      ]
    },
    shortcut: { type: String, default: void 0 },
    minDate: { type: String, default: void 0 },
    maxDate: { type: String, default: void 0 },
    disabledWeekly: {
      type: Array,
      default: () => [],
      validator: (e) => 7 >= e && e >= 0
    },
    disabledDates: { type: Array, default: () => [] },
    disabledHours: {
      type: Array,
      default: () => [],
      validator: (e) => 23 >= e && e >= 0
    }
  },
  emits: ["update:model-value", "close"],
  setup(e, { emit: u }) {
    const h = te(() => import("./chevron-down-bC0s_0yH.mjs")), y = te(() => import("./MazInput-BJnR_H4m.mjs"));
    p.extend(We), p.extend(Fe);
    const $ = {
      dateStyle: "full"
    }, t = e, k = Ve(), M = K(t.locale || "en-US"), T = R(() => t.locale ?? M.value), b = R(() => `mazPickerContainer-${k == null ? void 0 : k.uid}`), x = u, S = K(), W = R(() => t.time || t.onlyTime), J = R(() => t.double && !t.onlyTime), Q = R(() => !t.onlyTime), v = R(() => typeof m.value == "object");
    ze(() => {
      v.value && W.value && console.error("[maz-ui](MazPicker) You can't use time picker with range picker"), W.value && !(t.format.includes("h") || t.format.includes("H")) && console.error(
        '[maz-ui](MazPicker) When you use the time picker, you must provided a format with time - Ex: "YYYY-MM-DD HH:mm"'
      ), t.format.includes("h") && !(t.format.includes("a") || t.format.includes("A")) && console.error(
        '[maz-ui](MazPicker) if you use the 12 format "h" or "hh", you must add "a" or "A" at the end of the format - Ex: "YYYY-MM-DD hh:mm a"'
      );
    });
    const m = R({
      get: () => t.modelValue && typeof t.modelValue == "object" ? {
        start: t.modelValue.start ? p(t.modelValue.start, t.format).format() : void 0,
        end: t.modelValue.end ? p(t.modelValue.end, t.format).format() : void 0
      } : t.modelValue ? p(t.modelValue, t.format).format() : void 0,
      set: (o) => {
        t.disabled || (d(o), t.autoClose && o !== "object" && j());
      }
    }), O = K(((o) => {
      const a = (typeof o == "object" ? o.start : o) ?? p().format();
      return t.minDate && p(a).isBefore(t.minDate) ? t.minDate : t.maxDate && p(a).isAfter(t.maxDate) ? t.minDate ?? t.maxDate : a;
    })(m.value)), P = R(
      () => t.format.includes("a") || t.format.includes("A") || t.format.includes("h")
    ), A = R(() => ({
      ...$,
      ...t.inputDateStyle,
      timeStyle: t.inputDateStyle.timeStyle ?? W.value ? "short" : void 0,
      hour12: t.inputDateStyle.hour12 ?? P.value
    })), q = R(() => {
      if (m.value)
        return t.onlyTime ? m.value ? ve(p(m.value).format(), T.value, {
          timeStyle: A.value.timeStyle,
          hour12: A.value.hour12
        }) : void 0 : typeof m.value == "object" ? Ge({
          value: m.value,
          locale: T.value,
          options: A.value
        }) : he({
          value: p(m.value).format(),
          locale: T.value,
          options: A.value
        });
    }), C = K(!1), H = K(!1), z = K({
      vertical: "bottom",
      horizontal: "left"
    }), B = R(() => (C.value || t.open || H.value) && !t.disabled || t.inline), Y = K(!1);
    Be(async () => {
      if (Y.value = !0, t.customElementSelector && n(t.customElementSelector), !t.locale) {
        const o = et();
        if (!t.noUseBrowserLocale && o)
          M.value = o;
        else if (!t.noFetchLocal) {
          const a = await tt();
          a && (M.value = a);
        }
      }
    }), Ee(() => {
      t.customElementSelector && s(t.customElementSelector);
    });
    const D = async () => {
      if (t.pickerPosition) {
        const o = t.pickerPosition.includes("right") ? "right" : "left", a = t.pickerPosition.includes("top") ? "top" : "bottom";
        return {
          horizontal: o,
          vertical: a
        };
      } else
        return {
          horizontal: "left",
          vertical: await E(S.value)
        };
    }, E = async (o) => {
      if (typeof window > "u")
        return "bottom";
      const a = 30;
      await be();
      const f = document.querySelector(`#${b.value}`), w = o == null ? void 0 : o.getBoundingClientRect(), F = window.innerHeight, N = ((f == null ? void 0 : f.clientHeight) ?? 0) - a, Z = (w && F - w.bottom) ?? 0, I = (w && w.top) ?? 0, V = Z && Z >= N, U = I && I >= N;
      return !V && (U || I >= Z) ? "top" : "bottom";
    }, j = () => {
      C.value = !1, H.value = !1, x("close");
    };
    function c() {
      t.customElementSelector || j();
    }
    const i = () => {
      H.value = !H.value;
    }, n = (o) => {
      const a = document.querySelector(o);
      if (a)
        a.addEventListener("click", i);
      else
        throw new Error(
          `[maz-ui](MazPicker) impossible to find custom element with selector "${o}"`
        );
    }, s = (o) => {
      const a = document.querySelector(o);
      a == null || a.removeEventListener("click", i);
    }, r = (o) => {
      if (t.minDate || t.maxDate) {
        if (typeof o == "string") {
          const { newValue: a, newCurrentDate: f } = me({
            value: o,
            minDate: t.minDate,
            maxDate: t.maxDate,
            format: t.format
          });
          a && d(a), f && l(f);
        } else if (typeof o == "object" && (o.start || o.end)) {
          let a = o.start, f = o.end;
          if (o.start) {
            const { newValue: w, newCurrentDate: F } = me({
              value: o.start,
              minDate: t.minDate,
              maxDate: t.maxDate,
              format: t.format
            });
            w && (a = w), F && l(F);
          }
          if (o.end) {
            const { newValue: w } = me({
              value: o.end,
              minDate: t.minDate,
              maxDate: t.maxDate,
              format: t.format
            });
            w && (f = w);
          }
          d({
            start: a,
            end: f
          });
        }
      }
    }, l = (o) => {
      o && !p(O.value).isSame(o, "month") && (O.value = o);
    }, d = (o) => {
      if (typeof o == "object") {
        const a = _e(o, t.format);
        x("update:model-value", a), a.start && l(a.start);
      } else
        x("update:model-value", ye(o, t.format));
    };
    de(
      () => [m.value, t.minDate, t.maxDate],
      (o, a) => {
        const f = o[0], w = a == null ? void 0 : a[0];
        typeof f == "object" && (f.start || f.end) ? (!w || typeof w == "object" && (w.start !== f.start || w.end !== f.end)) && (d(f), r(f)) : typeof f == "string" && f !== w && (d(f), r(f));
      },
      { immediate: !0 }
    );
    function g(o) {
      o.code === "Escape" && B.value && (o.preventDefault(), j());
    }
    return de(
      () => B.value,
      async (o) => {
        o ? (z.value = await D(), !t.inline && Y.value && document.addEventListener("keydown", g)) : !t.inline && Y.value && document.removeEventListener("keydown", g);
      },
      { immediate: !0 }
    ), de(
      () => [m.value, t.disabledWeekly, t.disabledDates],
      (o) => {
        const a = o[0], f = o[1], w = o[2];
        typeof a == "object" && (a.start || a.end) ? ((a.start && ae({ value: a.start, disabledWeekly: f }) || a.start && ne({ value: a.start, disabledDates: w }) || a.end && ae({ value: a.end, disabledWeekly: f }) || a.end && ne({ value: a.end, disabledDates: w })) && (m.value = { start: void 0, end: void 0 }), (a.end && ae({ value: a.end, disabledWeekly: f }) || a.end && ne({ value: a.end, disabledDates: w })) && (m.value = { start: a.start, end: void 0 })) : typeof a == "string" && (ae({ value: a, disabledWeekly: f }) || ne({ value: a, disabledDates: w })) && (m.value = void 0);
      },
      { immediate: !0 }
    ), (o, a) => De((_(), we(
      "div",
      {
        ref_key: "MazPicker",
        ref: S,
        class: ge(["m-picker", [
          `m-picker--${e.color}`,
          `m-picker--${z.value.vertical}`,
          `m-picker--${z.value.horizontal}`,
          {
            "--is-open": B.value,
            "--is-disabled": e.disabled
          },
          t.class
        ]]),
        role: "none",
        style: Ae(e.style)
      },
      [
        !e.customElementSelector && !e.inline ? (_(), re(ee(y), He({
          key: 0,
          "model-value": q.value,
          readonly: ""
        }, o.$attrs, {
          autocomplete: "off",
          class: "m-picker__input",
          label: e.label,
          disabled: e.disabled,
          placeholder: e.placeholder,
          color: e.color,
          onClick: a[0] || (a[0] = (f) => C.value = !C.value)
        }), {
          "right-icon": pe(() => [
            $e("button", ot, [
              fe(ee(h), { class: "m-picker__button__chevron maz-text-lg" })
            ])
          ]),
          _: 1
          /* STABLE */
        }, 16, ["model-value", "label", "disabled", "placeholder", "color"])) : oe("v-if", !0),
        fe(Pe, {
          name: z.value.vertical === "top" ? "maz-slideinvert" : "maz-slide",
          persisted: ""
        }, {
          default: pe(() => [
            De(fe(rt, {
              id: b.value,
              ref: "PickerContainer",
              modelValue: m.value,
              "onUpdate:modelValue": a[1] || (a[1] = (f) => m.value = f),
              "calendar-date": O.value,
              "onUpdate:calendarDate": a[2] || (a[2] = (f) => O.value = f),
              "is-open": B.value,
              color: e.color,
              locale: T.value,
              "has-date": Q.value,
              double: J.value,
              "has-time": W.value,
              "formatter-options": A.value,
              "no-header": e.noHeader,
              "min-date": e.minDate,
              format: e.format,
              "is-hour12": P.value,
              "max-date": e.maxDate,
              "disabled-weekly": e.disabledWeekly,
              inline: e.inline,
              "first-day-of-week": e.firstDayOfWeek,
              shortcuts: e.shortcuts,
              shortcut: e.shortcut,
              disabled: e.disabled,
              "disabled-hours": e.disabledHours,
              "disabled-dates": e.disabledDates,
              "minute-interval": e.minuteInterval,
              "no-shortcuts": e.noShortcuts,
              onClose: j
            }, null, 8, ["id", "modelValue", "calendar-date", "is-open", "color", "locale", "has-date", "double", "has-time", "formatter-options", "no-header", "min-date", "format", "is-hour12", "max-date", "disabled-weekly", "inline", "first-day-of-week", "shortcuts", "shortcut", "disabled", "disabled-hours", "disabled-dates", "minute-interval", "no-shortcuts"]), [
              [je, B.value]
            ])
          ]),
          _: 1
          /* STABLE */
        }, 8, ["name"])
      ],
      6
      /* CLASS, STYLE */
    )), [
      [ee(Ze), c]
    ]);
  }
}), Dt = /* @__PURE__ */ Ce(it, [["__scopeId", "data-v-dc6be98f"]]);
export {
  Dt as M,
  Ce as _,
  ve as a,
  ct as b,
  Xe as c,
  p as d,
  mt as e,
  vt as f,
  ft as g,
  Ke as h,
  ht as i,
  dt as j,
  yt as s
};
