// maz-ui/components/chunks/MazStepper-NrorpacW.mjs
import "../assets/MazStepper.css";
import { defineComponent as L, useCssVars as q, defineAsyncComponent as m, computed as f, useSlots as G, ref as H, openBlock as c, createElementBlock as d, Fragment as z, renderList as J, normalizeClass as E, createElementVNode as r, createBlock as K, resolveDynamicComponent as Q, createCommentVNode as S, createTextVNode as h, toDisplayString as b, renderSlot as y, createVNode as R, unref as U, withCtx as X, withDirectives as Y, vShow as Z } from "vue";
const ee = { class: "m-stepper" }, te = ["disabled", "onClick"], se = { class: "m-stepper__header__wrapper" }, oe = { class: "m-stepper__count --primary" }, ae = { class: "m-stepper__count__circle" }, le = { class: "m-stepper__header__content" }, ne = { class: "m-stepper__title" }, re = {
  key: 0,
  class: "m-stepper__subtitle"
}, ce = {
  key: 0,
  class: "m-stepper__right"
}, ie = { class: "m-stepper__content__wrapper" }, ue = /* @__PURE__ */ L({
  __name: "MazStepper",
  props: {
    modelValue: { type: Number, default: void 0 },
    steps: { type: Array, default: void 0 },
    disabledNextSteps: { type: Boolean, default: !1 },
    disabledPreviousSteps: { type: Boolean, default: !1 },
    autoValidateSteps: { type: Boolean, default: !1 },
    allStepsOpened: { type: Boolean, default: !1 },
    allStepsValidated: { type: Boolean, default: !1 },
    color: {
      type: String,
      default: "primary"
    },
    canCloseSteps: { type: Boolean, default: !1 }
  },
  emits: ["update:model-value"],
  setup(i, { emit: C }) {
    q((e) => ({
      "34bce4d0": M.value,
      cb1aa1a0: P.value
    }));
    const _ = m(() => import("./MazTransitionExpand-53irAK2U.mjs")), g = m(() => import("./check-circle-JGd6LGss.mjs")), $ = m(
      () => import("./exclamation-circle-AoEMgPzE.mjs")
    ), A = m(() => import("./exclamation-triangle-KCpIguWV.mjs")), s = i, M = f(() => `var(--maz-color-${s.color})`), P = f(() => `var(--maz-color-${s.color}-contrast)`), W = C, I = G(), v = f(
      () => Object.keys(I).filter((e) => e.startsWith("content-")).length
    ), N = H(1), a = f({
      get: () => s.modelValue ?? N.value,
      set: (e) => {
        N.value = e, W("update:model-value", e);
      }
    }), k = (e) => O(e) ? { icon: g, class: "--success" } : w(e) ? { icon: A, class: "--warning" } : B(e) ? { icon: $, class: "--error" } : { class: "--normal" }, V = (e, o) => {
      var t, l;
      return (l = (t = s.steps) == null ? void 0 : t[o - 1]) == null ? void 0 : l[e];
    }, x = (e) => {
      a.value === e && s.canCloseSteps ? a.value = 0 : e < 1 ? a.value = 1 : e > v.value ? a.value = v.value : a.value = e;
    }, D = (e, o) => {
      var n, p;
      const t = e === "titleInfo" ? "title-info" : e, l = Object.keys(I).filter((F) => F.startsWith(`${t}-`)).includes(`${t}-${o}`), u = !!((p = (n = s.steps) == null ? void 0 : n[o - 1]) != null && p[e]);
      return l || u;
    }, O = (e) => {
      var u, n;
      const o = (n = (u = s.steps) == null ? void 0 : u[e - 1]) == null ? void 0 : n.success, t = B(e) || w(e), l = s.autoValidateSteps && e < a.value && !t;
      return o ?? (l || s.allStepsValidated);
    }, T = (e) => {
      var n, p;
      const o = (p = (n = s.steps) == null ? void 0 : n[e - 1]) == null ? void 0 : p.disabled, t = a.value === e && !s.canCloseSteps, l = s.disabledNextSteps && e > a.value, u = s.disabledPreviousSteps && e < a.value;
      return o ?? (t || l || u || s.allStepsOpened);
    }, B = (e) => {
      var o, t;
      return (t = (o = s.steps) == null ? void 0 : o[e - 1]) == null ? void 0 : t.error;
    }, w = (e) => {
      var o, t;
      return (t = (o = s.steps) == null ? void 0 : o[e - 1]) == null ? void 0 : t.warning;
    }, j = (e) => e === v.value;
    return (e, o) => (c(), d("div", ee, [
      (c(!0), d(
        z,
        null,
        J(v.value, (t) => (c(), d(
          z,
          { key: t },
          [
            D("title", t) ? (c(), d("button", {
              key: 0,
              type: "button",
              disabled: T(t),
              class: E(["m-stepper__header", [
                {
                  "--is-current-step": t === a.value || i.allStepsOpened
                },
                `${k(t).class}`
              ]]),
              onClick: (l) => x(t)
            }, [
              r("div", se, [
                r("span", oe, [
                  r("div", ae, [
                    k(t).icon ? (c(), K(Q(k(t).icon), {
                      key: 0,
                      class: "icon maz-text-xl"
                    })) : S("v-if", !0)
                  ]),
                  h(
                    " " + b(t),
                    1
                    /* TEXT */
                  )
                ]),
                r("div", le, [
                  r("span", ne, [
                    y(e.$slots, `title-${t}`, {}, () => [
                      h(
                        b(V("title", t)),
                        1
                        /* TEXT */
                      )
                    ], !0)
                  ]),
                  D("subtitle", t) ? (c(), d("span", re, [
                    y(e.$slots, `subtitle-${t}`, {}, () => [
                      h(
                        b(V("subtitle", t)),
                        1
                        /* TEXT */
                      )
                    ], !0)
                  ])) : S("v-if", !0)
                ])
              ]),
              D("titleInfo", t) ? (c(), d("span", ce, [
                y(e.$slots, `title-info-${t}`, {}, () => [
                  h(
                    b(V("titleInfo", t)),
                    1
                    /* TEXT */
                  )
                ], !0)
              ])) : S("v-if", !0)
            ], 10, te)) : S("v-if", !0),
            r(
              "div",
              {
                class: E(["m-stepper__content", {
                  "--no-border": j(t)
                }])
              },
              [
                R(
                  U(_),
                  null,
                  {
                    default: X(() => [
                      Y(r(
                        "div",
                        null,
                        [
                          r("div", ie, [
                            y(e.$slots, `content-${t}`, {
                              validated: O(t),
                              error: B(t),
                              warning: w(t),
                              nextStep: () => x(t + 1),
                              previousStep: () => x(t - 1)
                            }, void 0, !0)
                          ])
                        ],
                        512
                        /* NEED_PATCH */
                      ), [
                        [Z, i.allStepsOpened || a.value === t]
                      ])
                    ]),
                    _: 2
                    /* DYNAMIC */
                  },
                  1024
                  /* DYNAMIC_SLOTS */
                )
              ],
              2
              /* CLASS */
            )
          ],
          64
          /* STABLE_FRAGMENT */
        ))),
        128
        /* KEYED_FRAGMENT */
      ))
    ]));
  }
}), de = (i, C) => {
  const _ = i.__vccOpts || i;
  for (const [g, $] of C)
    _[g] = $;
  return _;
}, ve = /* @__PURE__ */ de(ue, [["__scopeId", "data-v-201c89f3"]]);
export {
  ve as M,
  de as _
};
