// maz-ui/modules/chunks/index-w7bAz52b.mjs
var Ge = Object.defineProperty;
var Qe = (s, e, t) => e in s ? Ge(s, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : s[e] = t;
var l = (s, e, t) => (Qe(s, typeof e != "symbol" ? e + "" : e, t), t);
import "../assets/index.css";
import { inject as Je, createVNode as $, render as ee, nextTick as He, defineComponent as $e, defineAsyncComponent as E, onBeforeMount as et, ref as h, computed as g, onMounted as Ee, onBeforeUnmount as tt, openBlock as L, createElementBlock as Z, normalizeClass as re, withModifiers as M, withKeys as st, withDirectives as be, unref as T, vShow as we, createCommentVNode as D, createElementVNode as I, Transition as De, withCtx as oe, watch as it, createBlock as U, resolveDynamicComponent as nt, toDisplayString as pe, createTextVNode as ot, normalizeStyle as rt } from "vue";
function Le(s, e) {
  const t = Je(s, e);
  if (!t)
    throw new TypeError(`[maz-ui](injectStrict) Could not resolve ${s}`);
  return t;
}
function as(s, e) {
  let t;
  return function(...i) {
    clearTimeout(t), t = setTimeout(() => {
      s.apply(this, i);
    }, e);
  };
}
function at(s) {
  return !!s;
}
function Me(s, e) {
  let t = e == null ? void 0 : e.element;
  function i() {
    t && ee(null, t);
  }
  const n = $(
    s,
    { ...e == null ? void 0 : e.props, ...e != null && e.addDestroyInProps ? { destroy: i } : {} },
    e == null ? void 0 : e.children
  );
  return e != null && e.app ? (n.appContext = e.app._context, t ? ee(n, t) : typeof document < "u" && (t = document.createElement("div"), ee(n, t))) : (t = t ?? document.body, ee(n, t)), { vNode: n, destroy: i, el: t };
}
function S() {
  return typeof document < "u";
}
class lt {
  constructor(e, t) {
    l(this, "defaultOptions", {
      element: void 0,
      timeout: 60 * 1e3 * 5,
      // 5 minutes
      once: !1,
      immediate: !0
    });
    l(this, "options");
    l(this, "timeoutHandler");
    l(this, "isIdle", !1);
    l(this, "isDestroy", !1);
    l(this, "startTime", 0);
    l(this, "remainingTime", 0);
    l(this, "lastClientX", -1);
    l(this, "lastClientY", -1);
    l(this, "eventNames", [
      "DOMMouseScroll",
      "mousedown",
      "mousemove",
      "mousewheel",
      "MSPointerDown",
      "MSPointerMove",
      "keydown",
      "touchmove",
      "touchstart",
      "wheel",
      "click"
    ]);
    l(this, "handleEvent", (e) => {
      try {
        if (this.remainingTime > 0)
          return;
        if (e.type === "mousemove") {
          const { clientX: t, clientY: i } = e;
          if (t === void 0 && i === void 0 || t === this.lastClientX && i === this.lastClientY)
            return;
          this.lastClientX = t, this.lastClientY = i;
        }
        this.resetTimeout(), this.callback({ isIdle: this.isIdle, eventType: e.type });
      } catch (t) {
        throw new Error(`[IdleTimeout](handleEvent) ${t}`);
      }
    });
    this.callback = e, this.options = {
      ...this.defaultOptions,
      ...t
    }, S() && this.start();
  }
  get element() {
    return this.options.element ?? document.body;
  }
  start() {
    if (!S()) {
      console.warn("[IdleTimeout](start) you should run this method on client side");
      return;
    }
    for (const e of this.eventNames)
      this.element.addEventListener(e, this.handleEvent);
    this.resetTimeout(), this.options.immediate && this.callback({ isIdle: !1 });
  }
  pause() {
    const e = this.startTime + this.options.timeout - Date.now();
    e <= 0 || (this.remainingTime = e, this.timeoutHandler && (clearTimeout(this.timeoutHandler), this.timeoutHandler = void 0));
  }
  resume() {
    this.remainingTime <= 0 || (this.resetTimeout(), this.callback({ isIdle: this.isIdle }), this.remainingTime = 0);
  }
  reset() {
    this.isDestroy = !1, this.isIdle = !1, this.remainingTime = 0, this.resetTimeout(), this.callback({ isIdle: this.isIdle });
  }
  destroy() {
    if (!S()) {
      console.warn("[IdleTimeout](destroy) you should run this method on client side");
      return;
    }
    this.isDestroy = !0;
    for (const e of this.eventNames)
      this.element.removeEventListener(e, this.handleEvent);
    this.timeoutHandler && clearTimeout(this.timeoutHandler);
  }
  resetTimeout() {
    this.isIdle = !1, this.timeoutHandler && (clearTimeout(this.timeoutHandler), this.timeoutHandler = void 0), this.timeoutHandler = setTimeout(
      this.handleTimeout.bind(this),
      this.remainingTime || this.options.timeout
    ), this.startTime = Date.now();
  }
  handleTimeout() {
    this.isIdle = !0, this.callback({ isIdle: this.isIdle }), this.options.once && this.destroy();
  }
  get destroyed() {
    return this.isDestroy;
  }
  get timeout() {
    return this.options.timeout;
  }
  set timeout(e) {
    this.options.timeout = e;
  }
  get idle() {
    return this.isIdle;
  }
  set idle(e) {
    e ? this.handleTimeout() : this.reset(), this.callback({ isIdle: this.isIdle });
  }
}
class ct {
  constructor(e, t) {
    l(this, "eventHandlerFunction");
    l(this, "event", "visibilitychange");
    l(this, "timeoutHandler");
    l(this, "options");
    l(this, "defaultOptions", {
      timeout: 5e3,
      once: !1,
      immediate: !0
    });
    l(this, "isVisible", !1);
    this.callback = e, this.options = {
      ...this.defaultOptions,
      ...t
    }, this.eventHandlerFunction = this.eventHandler.bind(this), S() && this.start();
  }
  start() {
    if (!S()) {
      console.warn("[UserVisibility](start) you should run this method on client side");
      return;
    }
    this.options.immediate && this.emitCallback(), this.addEventListener();
  }
  emitCallback() {
    this.isVisible = document.visibilityState === "visible", this.callback({ isVisible: this.isVisible }), this.options.once && this.destroy();
  }
  eventHandler() {
    document.visibilityState === "visible" && !this.isVisible ? (this.clearTimeout(), this.emitCallback()) : document.visibilityState === "hidden" && this.initTimeout();
  }
  clearTimeout() {
    this.timeoutHandler && (clearTimeout(this.timeoutHandler), this.timeoutHandler = void 0);
  }
  initTimeout() {
    this.clearTimeout(), this.timeoutHandler = setTimeout(this.emitCallback.bind(this), this.options.timeout);
  }
  addEventListener() {
    document.addEventListener(this.event, this.eventHandlerFunction);
  }
  removeEventListener() {
    document.removeEventListener(this.event, this.eventHandlerFunction);
  }
  destroy() {
    this.removeEventListener(), this.timeoutHandler && clearTimeout(this.timeoutHandler);
  }
}
const X = (s) => new Promise((e) => {
  setTimeout(e, s);
});
class ls {
  constructor({ src: e, identifier: t, once: i = !0, async: n = !0, defer: c = !0 }) {
    l(this, "src");
    l(this, "script");
    l(this, "once");
    l(this, "async");
    l(this, "defer");
    l(this, "identifier");
    if (typeof window > "u")
      throw new TypeError("[ScriptLoader]: Is supported only on browser side");
    if (!e)
      throw new Error('[ScriptLoader]: You should provide the attribut "src"');
    if (!t)
      throw new Error('[ScriptLoader]: You should provide the attribut "identifier"');
    this.src = e, this.identifier = t, this.once = i, this.async = n, this.defer = c;
  }
  removeTag(e) {
    var t;
    typeof e == "string" ? (t = document.head.querySelector(`[data-identifier="${e}"]`)) == null || t.remove() : e.remove();
  }
  load() {
    const e = window, t = document.head.querySelectorAll(`[data-identifier="${this.identifier}"]`);
    if (this.once && e[this.identifier] && t.length > 0)
      return this.script = e[this.identifier], Promise.resolve(this.script);
    if (!this.once && t.length > 0)
      for (const i of t)
        this.removeTag(i);
    return this.injectScript();
  }
  injectScript() {
    const e = window;
    return new Promise((t, i) => {
      try {
        const n = document.createElement("script");
        n.src = this.src, n.async = this.async, n.defer = this.defer, n.dataset.identifier = this.identifier, n.addEventListener("error", (c) => i(new Error(`[ScriptLoader](injectScript) ${c.message}`))), n.addEventListener("load", (c) => (this.script = c, e[this.identifier] = c, t(c))), document.head.append(n);
      } catch (n) {
        throw new Error(`[ScriptLoader](init) ${n}`);
      }
    });
  }
}
function cs(s) {
  return [...s].map((e) => {
    const t = e.codePointAt(0);
    return t ? t % 32 + 127461 : void 0;
  }).filter(at).map((e) => String.fromCodePoint(e)).join("");
}
const ae = "__vue_click_away__", Be = () => document.ontouchstart === null ? "touchstart" : "click", Fe = async (s, e) => {
  Ne(s);
  const t = e.instance, i = e.value, n = typeof i == "function";
  if (!n)
    throw new Error("[maz-ui](vClickOutside) the callback should be a function");
  await He(), s[ae] = (d) => {
    if ((!s || !s.contains(d.target)) && i && n)
      return i.call(t, d);
  };
  const c = Be();
  document.addEventListener(c, s[ae], !1);
}, Ne = (s) => {
  const e = Be();
  document.removeEventListener(e, s[ae], !1), delete s[ae];
}, ut = (s, e) => {
  e.value !== e.oldValue && Fe(s, e);
}, _e = {
  mounted: Fe,
  updated: ut,
  unmounted: Ne
}, us = {
  install: (s) => {
    s.directive("click-outside", _e);
  }
}, qe = (s, e, t, i) => {
  s.stopPropagation();
  const { handler: n, exclude: c } = t.value;
  let d = !1;
  if (c && c.length > 0)
    for (const a of c)
      d || (d = i.context.$refs[a].contains(s.target));
  !e.contains(s.target) && !d && i.context[n]();
}, Ue = () => document.ontouchstart === null ? "touchstart" : "click", mt = (s, e, t) => {
  const i = Ue();
  document.removeEventListener(
    i,
    (n) => qe(n, s, e, t)
  );
}, dt = (s, e, t) => {
  const i = Ue();
  document.addEventListener(
    i,
    (n) => qe(n, s, e, t)
  );
}, Ve = {
  mounted: dt,
  unmounted: mt
}, ms = {
  install: (s) => {
    s.directive("closable", Ve);
  }
}, ht = `
.maz-zoom-img {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 1rem;
  z-index: 1050;
  background-color: hsla(238, 15%, 40%, 0.7);
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.maz-zoom-img,
.maz-zoom-img * {
  box-sizing: border-box;
}

.maz-zoom-img .maz-zoom-img__wrapper {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  max-width: 100%;
  max-height: 100%;
  transition: all 300ms ease-in-out;
  opacity: 0;
  transform: scale(0.5);
}

.maz-zoom-img.maz-animate .maz-zoom-img__wrapper {
  opacity: 1;
  transform: scale(1);
}

.maz-zoom-img.maz-animate .maz-zoom-img__loader {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: hsla(238, 15%, 40%, 0.7);
  border-radius: 1rem;
  z-index: 2;
  min-width: 60px;
  min-height: 60px;
}
.maz-zoom-img.maz-animate .maz-zoom-img__loader[hidden] {
  display: none;
}

@-webkit-keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.maz-zoom-img.maz-animate .maz-zoom-img__loader__svg {
  animation: spin .6s linear infinite;
}

.maz-zoom-img img {
  max-width: 100%;
  max-height: 100%;
  min-width: 0;
  border-radius: 1rem;
}

.maz-zoom-img .maz-zoom-btn {
  margin: 0 auto;
  border: none;
  background-color: hsla(0, 0%, 7%, 0.5);
  box-shadow: 0 0 0.5rem 0 hsla(0, 0%, 0%, 0.2);
  height: 2.2rem;
  min-height: 2.2rem;
  width: 2.2rem;
  min-width: 2.2rem;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 2.2rem;
  cursor: pointer;
  flex: 0 0 auto;
  outline: none;
}

.maz-zoom-img .maz-zoom-btn svg {
  fill: white;
}

.maz-zoom-img .maz-zoom-btn.maz-zoom-btn--close {
  position: absolute;
  top: 0.5rem;
  right: 0.5rem;
  z-index: 1;
}

.maz-zoom-img .maz-zoom-btn.maz-zoom-btn--previous {
  position: absolute;
  left: 0.5rem;
  z-index: 1;
}

.maz-zoom-img .maz-zoom-btn.maz-zoom-btn--next {
  position: absolute;
  right: 0.5rem;
  z-index: 1;
}

.maz-zoom-img .maz-zoom-btn:hover {
  background-color: hsl(0, 0%, 0%);
}`, ke = {
  close: '<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0z" fill="none"/><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>',
  next: '<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0z" fill="none"/><path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"/></svg>',
  previous: '<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0z" fill="none"/><path d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"/></svg>',
  spinner: '<svg width="40px" height="40px" version="1.1" xmlns="http://www.w3.org/2000/svg" fill="currentColor" x="0px" y="0px" viewBox="0 0 50 50" xml:space="preserve" class="maz-zoom-img__loader__svg" data-v-6d1cb50c=""><path d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"></path></svg>'
};
class vt {
  constructor(e) {
    l(this, "options");
    l(this, "loader");
    l(this, "wrapper");
    l(this, "img");
    l(this, "keydownHandler");
    l(this, "onImgLoadedCallback");
    l(this, "buttonsAdded");
    l(this, "defaultOptions", {
      scale: !0,
      blur: !0,
      disabled: !1
    });
    l(this, "mouseEnterListener");
    l(this, "mouseLeaveListener");
    l(this, "renderPreviewListener");
    if (!e.value)
      throw new Error(
        '[MazUI](zoom-img) Image path must be defined. Ex: `v-zoom-img="<PATH_TO_IMAGE>"`'
      );
    if (typeof e.value == "object" && !e.value.src)
      throw new Error("[maz-ui](zoom-img) src of image must be provided");
    this.buttonsAdded = !1, this.options = this.buildOptions(e), this.keydownHandler = this.keydownLister.bind(this), this.loader = this.getLoader(), this.wrapper = document.createElement("div"), this.wrapper.classList.add("maz-zoom-img__wrapper"), this.wrapper.prepend(this.loader), this.img = document.createElement("img"), this.onImgLoadedCallback = this.onImgLoaded.bind(this), this.imgEventHandler(!0);
  }
  buildOptions(e) {
    return {
      ...this.defaultOptions,
      ...typeof e.value == "object" ? e.value : { src: e.value }
    };
  }
  get allInstances() {
    return [...document.querySelectorAll(".maz-zoom-img-instance")];
  }
  create(e) {
    this.options.disabled || (e.style.cursor = "pointer", setTimeout(() => e.classList.add("maz-zoom-img-instance")), e.setAttribute("data-zoom-src", this.options.src), this.options.alt && e.setAttribute("data-zoom-alt", this.options.alt), e.style.transition = "all 300ms ease-in-out", this.mouseEnterListener = () => this.mouseEnter(e), this.mouseLeaveListener = () => this.mouseLeave(e), this.renderPreviewListener = () => this.renderPreview(e, this.options), e.addEventListener("mouseenter", this.mouseEnterListener), e.addEventListener("mouseleave", this.mouseLeaveListener), e.addEventListener("click", this.renderPreviewListener));
  }
  update(e) {
    this.options = this.buildOptions(e);
  }
  remove(e) {
    this.imgEventHandler(!1), e.removeEventListener("mouseenter", this.mouseEnterListener), e.removeEventListener("mouseleave", this.mouseLeaveListener), e.removeEventListener("click", this.renderPreviewListener), e.classList.remove("maz-zoom-img-instance"), e.removeAttribute("data-zoom-src"), e.removeAttribute("data-zoom-alt"), e.style.cursor = "";
  }
  renderPreview(e, t) {
    e.classList.add(
      "maz-is-open"
      /* OPEN */
    ), this.addStyle(ht);
    const i = document.createElement("div");
    i.classList.add("maz-zoom-img"), i.setAttribute("id", "MazImgPreviewFullsize"), i.addEventListener("click", (n) => {
      i.isEqualNode(n.target) && this.closePreview();
    }), typeof t == "object" && (this.img.setAttribute("src", t.src), t.alt && this.img.setAttribute("alt", t.alt), this.img.id = "MazImgElement"), this.wrapper.append(this.img), i.append(this.wrapper), document.body.append(i), this.keyboardEventHandler(!0), setTimeout(() => {
      i && i.classList.add("maz-animate");
    }, 100);
  }
  onImgLoaded() {
    this.wrapper.style.width = `${this.img.width}px`, this.wrapper.style.minWidth = "200px", this.loader.hidden = !0;
    const e = this.getButton(), t = [], i = this.allInstances.length > 1;
    if (!this.buttonsAdded) {
      if (this.buttonsAdded = !0, i) {
        const n = this.getButton("previous"), c = this.getButton("next");
        t.push(n, c);
      }
      this.wrapper.append(e), i && (this.wrapper.prepend(t[0]), this.wrapper.append(t[1]));
    }
  }
  getLoader() {
    const e = document.createElement("div");
    return e.classList.add("maz-zoom-img__loader"), e.innerHTML = ke.spinner, e;
  }
  mouseLeave(e) {
    this.options.scale && (e.style.transform = ""), this.options.blur && (e.style.filter = ""), e.style.zIndex = "";
  }
  mouseEnter(e) {
    e.style.zIndex = "1", this.options.scale && (e.style.transform = "scale(1.1)"), this.options.blur && (e.style.filter = "blur(2px)");
  }
  keydownLister(e) {
    e.preventDefault(), (e.key === "Escape" || e.key === " ") && this.closePreview(), (e.key === "ArrowLeft" || e.key === "ArrowRight") && this.nextPreviousImage(e.key === "ArrowRight");
  }
  getButton(e = "close") {
    const t = document.createElement("button");
    return t.innerHTML = ke[e], t.addEventListener("click", () => {
      e === "close" ? this.closePreview() : this.allInstances && this.nextPreviousImage(e === "next");
    }), t.classList.add("maz-zoom-btn"), t.classList.add(`maz-zoom-btn--${e}`), t;
  }
  closePreview() {
    const e = document.querySelector("#MazImgPreviewFullsize"), t = document.querySelector("#MazPreviewStyle"), i = document.querySelector(
      ".maz-zoom-img-instance.maz-is-open"
    );
    i && i.classList.remove(
      "maz-is-open"
      /* OPEN */
    ), e && e.classList.remove("maz-animate"), this.keyboardEventHandler(!1), setTimeout(() => {
      e && e.remove(), t && t.remove();
    }, 300);
  }
  getNewInstanceIndex(e) {
    return e < 0 ? this.allInstances.length - 1 : e >= this.allInstances.length ? 0 : e;
  }
  nextPreviousImage(e) {
    const t = e, i = document.querySelector(
      ".maz-zoom-img-instance.maz-is-open"
    );
    if (i) {
      const n = this.allInstances.indexOf(i), c = t ? n + 1 : n - 1, d = this.allInstances[this.getNewInstanceIndex(c)];
      d && this.useNextInstance(i, d);
    }
  }
  useNextInstance(e, t) {
    e.classList.remove(
      "maz-is-open"
      /* OPEN */
    ), t.classList.add(
      "maz-is-open"
      /* OPEN */
    );
    const i = t.getAttribute("data-zoom-src"), n = t.getAttribute("data-zoom-alt");
    this.wrapper.style.width = "", this.loader.hidden = !1, i && this.img.setAttribute("src", i), n && this.img.setAttribute("alt", n);
  }
  addStyle(e) {
    const t = document.createElement("style");
    t.id = "MazPreviewStyle", t.textContent = e, document.head.append(t);
  }
  keyboardEventHandler(e) {
    if (e)
      return document.addEventListener("keydown", this.keydownHandler);
    document.removeEventListener("keydown", this.keydownHandler);
  }
  imgEventHandler(e) {
    if (e)
      return this.img.addEventListener("load", this.onImgLoadedCallback);
    this.img.removeEventListener("load", this.onImgLoadedCallback);
  }
}
let te;
const je = {
  created(s, e) {
    te = new vt(e), te.create(s);
  },
  updated(s, e) {
    te.update(e);
  },
  unmounted(s) {
    te.remove(s);
  }
}, ds = {
  install(s) {
    s.directive("zoom-img", je);
  }
}, ft = "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7", ze = {
  baseClass: "m-lazy-img",
  loadedClass: "m-lazy-loaded",
  loadingClass: "m-lazy-loading",
  errorClass: "m-lazy-error",
  noPhotoClass: "m-lazy-no-photo",
  noPhoto: !1,
  observerOnce: !0,
  loadOnce: !1,
  noUseErrorPhoto: !1,
  observerOptions: {
    threshold: 0.1
  }
};
class We {
  constructor(e = {}) {
    l(this, "observers", []);
    l(this, "defaultOptions", ze);
    l(this, "options");
    l(this, "onImgLoadedCallback");
    l(this, "onImgErrorCallback");
    l(this, "hasImgLoaded", !1);
    this.options = this.buildOptions(e), this.onImgLoadedCallback = this.imageIsLoaded.bind(this), this.onImgErrorCallback = this.imageHasError.bind(this);
  }
  async loadErrorPhoto() {
    const { default: e } = await import("./no-photography-1qnDtpgo.mjs");
    return e;
  }
  buildOptions(e) {
    return {
      ...this.defaultOptions,
      ...e,
      observerOptions: {
        ...this.defaultOptions.observerOptions,
        ...e.observerOptions
      }
    };
  }
  removeClass(e, t) {
    e.classList.remove(t);
  }
  addClass(e, t) {
    e.classList.add(t);
  }
  removeAllStateClasses(e) {
    this.removeClass(e, this.options.loadedClass), this.removeClass(e, this.options.loadingClass), this.removeClass(e, this.options.errorClass), this.removeClass(e, this.options.noPhotoClass);
  }
  setBaseClass(e) {
    this.addClass(e, this.options.baseClass);
  }
  imageIsLoading(e) {
    var t, i;
    this.addClass(e, this.options.loadingClass), (i = (t = this.options).onLoading) == null || i.call(t, e);
  }
  imageHasNoPhoto(e) {
    this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.noPhotoClass), this.setDefaultPhoto(e);
  }
  imageIsLoaded(e) {
    var t, i;
    this.hasImgLoaded = !0, this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.loadedClass), (i = (t = this.options).onLoaded) == null || i.call(t, e);
  }
  imageHasError(e, t) {
    var i, n;
    console.warn("[maz-ui][MazLazyImg] Error while loading image", t), this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.errorClass), (n = (i = this.options).onError) == null || n.call(i, e), this.setDefaultPhoto(e);
  }
  getImageUrl(e, t) {
    const i = this.getImgElement(e).getAttribute("data-lazy-src");
    if (i)
      return i;
    t.value;
    const n = typeof t.value == "object" ? t.value.src : t.value;
    return n || console.warn("[maz-ui][MazLazyImg] src url is not defined"), n;
  }
  async setPictureSourceUrls(e) {
    const t = e.querySelectorAll("source");
    if (t.length > 0)
      for await (const i of t) {
        const n = i.getAttribute("data-lazy-srcset");
        n ? i.srcset = n : console.warn(
          '[maz-ui][MazLazyImg] the "[data-lazy-srcset]" attribute is not provided on "<source />"'
        );
      }
    else
      console.warn(
        '[maz-ui][MazLazyImg] No "<source />" elements provided into the "<picture />" element'
      ), this.imageHasError(e);
  }
  hasBgImgMode(e) {
    return e.arg === "bg-image";
  }
  isPictureElement(e) {
    return e instanceof HTMLPictureElement;
  }
  getImgElement(e) {
    return this.isPictureElement(e) ? e.querySelector("img") : e;
  }
  async setDefaultPhoto(e) {
    if (this.options.noUseErrorPhoto)
      return;
    const t = this.options.errorPhoto ?? await this.loadErrorPhoto(), i = e.querySelectorAll("source");
    if (i.length > 0)
      for await (const n of i)
        n.srcset = t;
    else
      this.setImgSrc(e, t);
  }
  addEventListenerToImg(e) {
    const t = this.getImgElement(e);
    t.addEventListener("load", () => this.onImgLoadedCallback(e), {
      once: !0
    }), t.addEventListener("error", (i) => this.onImgErrorCallback(e, i), { once: !0 });
  }
  async loadImage(e, t) {
    if (this.imageIsLoading(e), this.isPictureElement(e))
      this.addEventListenerToImg(e), await this.setPictureSourceUrls(e);
    else {
      const i = this.getImageUrl(e, t);
      if (!i)
        return this.imageHasError(e);
      this.hasBgImgMode(t) ? (e.style.backgroundImage = `url('${i}')`, this.imageIsLoaded(e)) : (this.addEventListenerToImg(e), this.setImgSrc(e, i));
    }
  }
  setImgSrc(e, t) {
    const i = this.getImgElement(e);
    i.src = t;
  }
  handleIntersectionObserver(e, t, i, n) {
    var c, d;
    this.observers.push(n);
    for (const a of i)
      if (a.isIntersecting) {
        if ((d = (c = this.options).onIntersecting) == null || d.call(c, a.target), this.options.observerOnce && n.unobserve(e), this.options.loadOnce && this.hasImgLoaded)
          return;
        this.loadImage(e, t);
      }
  }
  createObserver(e, t) {
    const i = (d, a) => {
      this.handleIntersectionObserver(e, t, d, a);
    }, n = this.options.observerOptions;
    new IntersectionObserver(i, n).observe(e);
  }
  async imageHandler(e, t, i) {
    if (i === "update")
      for await (const n of this.observers)
        n.unobserve(e);
    window.IntersectionObserver ? this.createObserver(e, t) : this.loadImage(e, t);
  }
  async bindUpdateHandler(e, t, i) {
    if (this.options.noPhoto)
      return this.imageHasNoPhoto(e);
    await this.imageHandler(e, t, i);
  }
  async add(e, t) {
    if (this.hasBgImgMode(t) && this.isPictureElement(e))
      throw new Error(`[MazLazyImg] You can't use the "bg-image" mode with "<picture />" element`);
    setTimeout(() => this.setBaseClass(e), 0), e.getAttribute("src") || this.setImgSrc(e, ft), await this.bindUpdateHandler(e, t, "bind");
  }
  async update(e, t) {
    t.value !== t.oldValue && (this.hasImgLoaded = !1, this.removeAllStateClasses(e), await this.bindUpdateHandler(e, t, "update"));
  }
  remove(e, t) {
    this.hasImgLoaded = !1, this.hasBgImgMode(t) && (e.style.backgroundImage = ""), this.removeAllStateClasses(e);
    for (const i of this.observers)
      i.unobserve(e);
    this.observers = [];
  }
}
let se;
const gt = {
  created(s, e) {
    const t = typeof e.value == "object" ? e.value : {};
    se = new We(t), se.add(s, e);
  },
  updated(s, e) {
    se.update(s, e);
  },
  unmounted(s, e) {
    se.remove(s, e);
  }
}, pt = "lazy-img", hs = {
  install(s, e = {}) {
    const t = {
      ...ze,
      ...e,
      observerOptions: {
        ...ze.observerOptions,
        ...e.observerOptions
      }
    }, i = new We(t);
    s.directive(pt, {
      created: i.add.bind(i),
      updated: i.update.bind(i),
      unmounted: i.remove.bind(i)
    });
  }
}, yt = ["onClick", "onKeypress"], bt = { class: "m-fullscreen-img-scroller" }, wt = ["src", "alt"], zt = /* @__PURE__ */ $e({
  __name: "MazFullscreenImg",
  props: {
    destroy: { type: Function, default: void 0 },
    src: {},
    alt: { default: void 0 },
    openInstanceClass: {},
    zoom: { type: Boolean, default: !0 },
    disabled: { type: Boolean },
    clickedElementBounds: {},
    clickedElement: {},
    animated: { type: Boolean, default: !0 },
    scaleAnimation: { type: Boolean }
  },
  emits: ["close", "previous", "next", "before-close"],
  setup(s, { emit: e }) {
    const t = e, i = E(() => import("./MazSpinner-CKpey7-9.mjs")), n = E(() => import("./x-mark-pPwcH6Hz.mjs")), c = E(() => import("./chevron-left-48sF6neb.mjs"));
    let d;
    et(async () => {
      d = await import("vue-scrollto");
    });
    const a = s, v = h(!1), p = h(!1), y = h(!1), f = h(!1), b = h(!1), w = h(!1), k = h(!1), O = h(a.clickedElement instanceof HTMLImageElement), G = h(a.src), B = h(a.alt), F = h(), u = h(), A = g(
      () => O.value && a.animated && !a.scaleAnimation
    ), N = g(
      () => (O.value || !a.animated) && !a.scaleAnimation
    ), _ = h(!1), le = g(() => w.value ? `--is-zoomed maz-cursor-zoom-in maz-max-w-[none] ${_.value ? "maz-h-screen maz-w-auto" : "maz-h-auto maz-w-screen"}` : `cursor-zoom-out ${_.value ? "maz-w-screen maz-h-screen --is-landscape" : "maz-h-screen maz-w-auto"}`);
    function ce() {
      p.value = !0, y.value = !1, f.value = !0;
    }
    function q() {
      t("before-close"), v.value = !1;
    }
    function Q(r) {
      r.key === "Escape" && (r.preventDefault(), q()), (r.key === "ArrowLeft" || r.key === "ArrowRight") && (r.preventDefault(), P(r.key === "ArrowRight"));
    }
    function ue() {
      document.documentElement.classList.add("--m-fullscreen-open");
    }
    function me() {
      document.documentElement.classList.remove("--m-fullscreen-open");
    }
    function R() {
      return [...document.querySelectorAll(".m-fullscreen-img-instance")];
    }
    function J(r, o) {
      return o < 0 ? r.length - 1 : o >= r.length ? 0 : o;
    }
    async function de(r, o) {
      r.classList.remove(a.openInstanceClass), o.classList.add(a.openInstanceClass);
      const m = o.getAttribute("data-src"), z = o.getAttribute("data-alt");
      B.value = z ?? void 0, m && (G.value = m), p.value = !1, await X(300), p.value || (y.value = !0);
    }
    function P(r) {
      const o = document.querySelector(
        `.m-fullscreen-img-instance.${a.openInstanceClass}`
      );
      if (o) {
        const m = R(), z = m.indexOf(o), H = r ? z + 1 : z - 1, x = m[J(m, H)];
        x && de(o, x), t(r ? "next" : "previous");
      }
    }
    async function he(r) {
      const o = r.target;
      o.style.width = `${o.clientWidth}px`, o.style.height = `${o.clientHeight}px`, await X(0), w.value = !w.value, o.style.removeProperty("width"), o.style.removeProperty("height"), await X(0);
      const m = o.clientWidth / r.clientY, H = window.innerWidth / m - window.innerHeight / 2;
      d.scrollTo(o, 200, {
        container: o.parentElement,
        easing: "linear",
        offset: w.value ? H : 0
      });
    }
    Ee(async () => {
      await He(), document.addEventListener("keydown", Q), v.value = !0, y.value = !0, ue(), b.value = R().length > 1, u.value && (_.value = a.clickedElementBounds.height < a.clickedElementBounds.width);
    });
    function K() {
      u.value && (k.value = !0, u.value.style.removeProperty("margin-left"), u.value.style.removeProperty("margin-right"), u.value.style.opacity = "0.5", u.value.style.width = `${a.clickedElementBounds.width}px`, u.value.style.height = `${a.clickedElementBounds.height}px`, u.value.style.top = `${a.clickedElementBounds.top}px`, u.value.style.left = `${a.clickedElementBounds.left}px`);
    }
    async function ve() {
      if (await X(0), u.value) {
        u.value.style.opacity = "1", u.value.style.removeProperty("height"), u.value.style.removeProperty("width"), u.value.style.removeProperty("transform");
        const r = window.innerWidth, o = window.innerHeight, m = a.clickedElementBounds.height, z = a.clickedElementBounds.width;
        if (_.value) {
          u.value.style.left = "0px";
          const H = m * r / z, x = (o - H) / 2;
          u.value.style.top = `${x <= 0 ? 0 : x}px`;
        } else {
          u.value.style.top = "0px";
          const H = z * o / m, x = (r - H) / 2;
          u.value.style.left = `${x <= 0 ? 0 : x}px`;
        }
      }
    }
    function Y() {
      u.value && (u.value.style.removeProperty("top"), u.value.style.removeProperty("left"), u.value.style.marginLeft = "auto", u.value.style.marginRight = "auto", u.value.style.removeProperty("opacity")), k.value = !1;
    }
    function fe() {
      if (u.value) {
        const { left: r, top: o } = u.value.getBoundingClientRect();
        u.value.style.left = `${r}px`, u.value.style.top = `${o}px`, u.value.style.width = `${u.value.width}px`, u.value.style.height = `${u.value.height}px`, k.value = !0;
      }
    }
    function ge() {
      var r, o;
      t("close"), (r = F.value) == null || r.remove(), (o = a.destroy) == null || o.call(a);
    }
    return tt(() => {
      document.removeEventListener("keydown", Q), me();
    }), (r, o) => (L(), Z("div", {
      ref_key: "FullscreenImgElement",
      ref: F,
      role: "button",
      class: re(["m-fullscreen-img", { "--animation-active": k.value }]),
      tabindex: "0",
      onClick: M(q, ["self"]),
      onKeypress: st(M(q, ["prevent"]), ["esc"])
    }, [
      be($(
        T(i),
        { class: "m-fullscreen-img-loader maz-h-16" },
        null,
        512
        /* NEED_PATCH */
      ), [
        [we, y.value]
      ]),
      f.value && b.value ? (L(), Z("button", {
        key: 0,
        type: "button",
        class: "m-fullscreen-btn --next",
        onClick: o[0] || (o[0] = M((m) => P(!0), ["stop"]))
      }, [
        $(T(c), { class: "maz-rotate-90 maz-text-2xl" })
      ])) : D("v-if", !0),
      v.value && f.value && b.value ? (L(), Z("button", {
        key: 1,
        type: "button",
        class: "m-fullscreen-btn --previous",
        onClick: o[1] || (o[1] = M((m) => P(!1), ["stop"]))
      }, [
        $(T(c), { class: "maz-text-2xl" })
      ])) : D("v-if", !0),
      I("button", {
        type: "button",
        class: "m-fullscreen-btn --close",
        onClick: q
      }, [
        $(T(n), { class: "maz-text-2xl" })
      ]),
      I("div", bt, [
        $(De, {
          name: N.value ? void 0 : "maz-scale-fade",
          onBeforeEnter: o[3] || (o[3] = (m) => A.value && K()),
          onEnter: o[4] || (o[4] = (m) => A.value && ve()),
          onAfterEnter: o[5] || (o[5] = (m) => A.value && Y()),
          onBeforeLeave: o[6] || (o[6] = (m) => A.value && fe()),
          onLeave: o[7] || (o[7] = (m) => A.value && K()),
          onAfterLeave: ge,
          persisted: ""
        }, {
          default: oe(() => [
            be(I("img", {
              id: "ZvImgElement",
              ref_key: "ImgElement",
              ref: u,
              src: G.value,
              alt: B.value,
              tabindex: "0",
              class: re([{ zoomable: r.zoom }, le.value]),
              onLoad: ce,
              onClick: o[2] || (o[2] = (m) => r.zoom ? he(m) : void 0)
            }, null, 42, wt), [
              [we, v.value]
            ])
          ]),
          _: 1
          /* STABLE */
        }, 8, ["name"])
      ])
    ], 42, yt));
  }
}), Re = (s, e) => {
  const t = s.__vccOpts || s;
  for (const [i, n] of e)
    t[i] = n;
  return t;
}, Et = /* @__PURE__ */ Re(zt, [["__scopeId", "data-v-d0c959c7"]]), Ae = "m-fullscreen-is-open";
class Lt {
  constructor() {
    l(this, "options");
    l(this, "defaultOptions", {
      scaleOnHover: !1,
      blurOnHover: !1,
      disabled: !1,
      animated: !0,
      zoom: !0,
      scaleAnimation: !1
    });
    l(this, "mouseEnterListener");
    l(this, "mouseLeaveListener");
    l(this, "renderPreviewListener");
  }
  buildOptions(e, t) {
    const i = typeof t.value == "object" ? t.value : { src: t.value, alt: void 0 }, n = (i == null ? void 0 : i.src) ?? this.getImgSrc(e), c = (i == null ? void 0 : i.alt) ?? this.getImgAlt(e);
    return {
      ...this.defaultOptions,
      ...i,
      src: n,
      alt: c
    };
  }
  get allInstances() {
    return [...document.querySelectorAll(".m-fullscreen-img-instance")];
  }
  getImgSrc(e) {
    var i;
    const t = ((i = this.options) == null ? void 0 : i.src) || e.getAttribute("src") || e.getAttribute("data-src");
    if (!t)
      throw new Error(
        '[maz-ui](fullscreen-img) src of image must be provided by `v-fullscreen=""`, `v-fullscreen="{ src: "" }"`, `src=""` or `data-src=""` atributes'
      );
    return t;
  }
  getImgAlt(e) {
    var t;
    return ((t = this.options) == null ? void 0 : t.alt) || e.getAttribute("alt") || e.getAttribute("data-alt");
  }
  create(e, t) {
    if (this.options = this.buildOptions(e, t), this.options.disabled)
      return;
    e.style.cursor = "move", (this.options.scaleOnHover || this.options.blurOnHover) && (e.style.transition = "all 300ms ease-in-out"), e.classList.add("m-fullscreen-img-instance"), e.setAttribute("data-src", this.getImgSrc(e));
    const i = this.getImgAlt(e);
    i && e.setAttribute("data-alt", i), this.mouseEnterListener = () => this.mouseEnter(e), this.mouseLeaveListener = () => this.mouseLeave(e), this.renderPreviewListener = () => this.renderPreview(e), e.addEventListener("mouseenter", this.mouseEnterListener), e.addEventListener("mouseleave", this.mouseLeaveListener), e.addEventListener("click", this.renderPreviewListener);
  }
  update(e, t) {
    this.options = this.buildOptions(e, t);
  }
  remove(e) {
    e.removeEventListener("mouseenter", this.mouseEnterListener), e.removeEventListener("mouseleave", this.mouseLeaveListener), e.removeEventListener("click", this.renderPreviewListener), e.classList.remove("m-fullscreen-img-instance"), e.style.cursor = "";
  }
  renderPreview(e) {
    return e.classList.add(Ae), Me(Et, {
      props: {
        ...this.options,
        openInstanceClass: Ae,
        clickedElement: e,
        clickedElementBounds: e.getBoundingClientRect()
      },
      addDestroyInProps: !0
    });
  }
  mouseLeave(e) {
    this.options.scaleOnHover && (e.style.transform = ""), this.options.blurOnHover && (e.style.filter = ""), e.style.zIndex = "";
  }
  mouseEnter(e) {
    e.style.zIndex = "1", this.options.scaleOnHover && (e.style.transform = "scale(1.05)"), this.options.blurOnHover && (e.style.filter = "blur(3px)");
  }
}
let ie;
const Ke = {
  mounted(s, e) {
    return ie = new Lt(), ie.create(s, e);
  },
  updated(s, e) {
    return ie.update(s, e);
  },
  unmounted(s) {
    return ie.remove(s);
  }
}, vs = {
  install(s) {
    s.directive("fullscreen-img", Ke);
  }
}, It = [
  {
    name: "click-outside",
    directive: _e
  },
  {
    name: "closable",
    directive: Ve
  },
  {
    name: "zoom-img",
    directive: je
  },
  {
    name: "lazy-img",
    directive: gt
  },
  {
    name: "fullscreen-img",
    directive: Ke
  }
], fs = {
  install(s) {
    for (const { name: e, directive: t } of It)
      s.directive(e, t);
  }
}, Tt = (s) => s ? (s = s.toString(), s.charAt(0).toUpperCase() + s.slice(1)) : "", kt = {
  style: "currency",
  minimumFractionDigits: 2,
  round: !1
}, At = (s, e, t) => {
  let i = Number(s);
  return t.round && (i = Math.floor(i), t.minimumFractionDigits = 0), new Intl.NumberFormat(e, t).format(i);
}, xt = (s, e, t) => {
  if (s === void 0)
    throw new TypeError("[maz-ui](FilterCurrency) The `number` attribute is required.");
  if (e === void 0)
    throw new TypeError("[maz-ui](FilterCurrency) The `locale` attribute is required.");
  if (typeof e != "string")
    throw new TypeError("[maz-ui](FilterCurrency) The `locale` attribute must be a string.");
  if (t.currency === void 0)
    throw new TypeError("[maz-ui](FilterCurrency) The `options.currency` attribute is required.");
}, Ct = (s, e, t) => {
  const i = {
    ...kt,
    ...t
  };
  xt(s, e, i);
  try {
    return At(s, e, i);
  } catch (n) {
    throw new Error(`[maz-ui](FilterCurrency) ${n}`);
  }
}, St = {
  month: "short",
  day: "numeric",
  year: "numeric"
}, Ot = (s, e, t) => {
  if (e === void 0)
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute is required.");
  if (typeof e != "string")
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute must be a string.");
  const i = t ?? St;
  try {
    const n = s instanceof Date ? s : new Date(s);
    return new Intl.DateTimeFormat(e, i).format(n);
  } catch (n) {
    throw new Error(`[maz-ui](FilterDate) ${n}`);
  }
}, Pt = {
  minimumFractionDigits: 2
}, Ht = (s, e, t) => {
  const i = {
    ...Pt,
    ...t
  };
  if (s === void 0)
    throw new TypeError("[maz-ui](FilterNumber) The `number` attribute is required.");
  if (e === void 0)
    throw new TypeError("[maz-ui](FilterNumber) The `locale` attribute is required.");
  if (typeof e != "string")
    throw new TypeError("[maz-ui](FilterNumber) The `locale` attribute must be a string.");
  try {
    return new Intl.NumberFormat(e, i).format(Number(s));
  } catch (n) {
    throw new Error(`[maz-ui](FilterNumber) ${n}`);
  }
}, $t = {
  capitalize: Tt,
  currency: Ct,
  date: Ot,
  number: Ht
}, gs = {
  install(s) {
    s.provide("filters", $t);
  }
}, Dt = ({ timeout: s, callback: e, remainingTimeUpdate: t = 200 }) => {
  const i = h(s);
  let n;
  function c() {
    i.value = s, d();
  }
  function d() {
    n || (n = setInterval(() => {
      i.value -= t, i.value <= 0 && (p(), e == null || e());
    }, t));
  }
  function a() {
    n && (clearInterval(n), n = void 0);
  }
  function v() {
    !n && i.value > 0 && d();
  }
  function p() {
    i.value = s, a();
  }
  return {
    remainingTime: i,
    start: c,
    pause: a,
    resume: v,
    stop: p
  };
}, Mt = { class: "m-toast__message-wrapper" }, Bt = { class: "m-toast__message" }, Ft = { class: "maz-flex maz-items-center maz-gap-2" }, Nt = { key: 0 }, _t = { class: "progress-bar maz-absolute maz-inset-x-0 maz-bottom-0 maz-h-1" }, qt = /* @__PURE__ */ $e({
  __name: "MazToast",
  props: {
    position: {
      type: String,
      default: "bottom-right"
    },
    maxToasts: { type: [Number, Boolean], default: !1 },
    timeout: { type: Number, default: 1e4 },
    queue: { type: Boolean, default: !1 },
    noPauseOnHover: { type: Boolean, default: !1 },
    type: {
      type: String,
      default: "info"
    },
    message: { type: String, required: !0 },
    link: { type: Object, default: void 0 },
    action: { type: Object, default: void 0 },
    persistent: { type: Boolean, default: !1 }
  },
  emits: ["close", "click", "open"],
  setup(s, { emit: e }) {
    const t = E(() => import("./MazBtn-23G6MO1u.mjs")), i = E(() => import("./x-mark-pPwcH6Hz.mjs")), n = E(
      () => import("./arrow-top-right-on-square-JGWhrxDH.mjs")
    ), c = E(
      () => import("./exclamation-triangle-KCpIguWV.mjs")
    ), d = E(
      () => import("./exclamation-circle-AoEMgPzE.mjs")
    ), a = E(
      () => import("./information-circle-5aSd26ke.mjs")
    ), v = E(() => import("./check-circle-JGd6LGss.mjs")), p = E(() => import("./link-CVsbXQMi.mjs")), y = h(), f = s, b = g(() => {
      switch (f.type) {
        case "danger":
          return c;
        case "info":
          return a;
        case "success":
          return v;
        case "warning":
          return d;
        default:
          return;
      }
    }), w = e, k = g(() => f.position.includes("top") ? "top" : "bottom"), O = g(() => f.position.includes("left") ? "left" : f.position.includes("right") ? "right" : "center"), G = g(() => O.value !== "center" ? O.value === "right" ? "m-slide-right" : "m-slide-left" : k.value === "top" ? "m-slide-top" : "m-slide-bottom"), B = h(!1), F = h(!1), u = h(), A = `m-toast-container --${k.value} --${O.value}`, N = `.${A.replaceAll(" ", ".")}`, { start: _, stop: le, pause: ce, resume: q, remainingTime: Q } = Dt({
      callback: Y,
      timeout: f.timeout
    });
    function ue() {
      const r = document.querySelector(N);
      if (!r && !r) {
        const o = document.body, m = document.createElement("div");
        m.className = A, o.append(m);
      }
    }
    function me() {
      const r = document.querySelector(N);
      return !f.queue && f.maxToasts === !1 ? !1 : typeof f.maxToasts == "number" && r ? f.maxToasts <= r.childElementCount : r && r.childElementCount > 0;
    }
    function R() {
      if (me()) {
        u.value = setTimeout(R, 250);
        return;
      }
      const r = document.querySelector(N);
      y.value && r && r.prepend(y.value), F.value = !0, _();
    }
    const J = h("100%");
    function de() {
      switch (f.type) {
        case "danger":
          return "maz-bg-danger-700";
        case "info":
          return "maz-bg-info-700";
        case "success":
          return "maz-bg-success-700";
        case "warning":
          return "maz-bg-warning-700";
        default:
          return "maz-bg-secondary";
      }
    }
    it(
      () => Q.value,
      (r) => {
        if (typeof r == "number") {
          const o = 100 * r / f.timeout;
          J.value = `${o}%`, r <= 0 && Y();
        }
      }
    );
    function P(r) {
      w("click", r), f.persistent || Y();
    }
    async function he(r, o) {
      var m;
      B.value = !0, await r(), B.value = !1, (m = f.action) != null && m.closeToast && P(o);
    }
    function K(r) {
      f.noPauseOnHover || (r ? ce() : q());
    }
    function ve() {
      le(), u.value && clearTimeout(u.value);
    }
    function Y() {
      ve(), F.value = !1;
    }
    function fe() {
      w("open");
    }
    function ge() {
      var o;
      w("close"), (o = y.value) == null || o.remove();
      const r = document.querySelector(N);
      r && !(r != null && r.hasChildNodes()) && r.remove();
    }
    return Ee(() => {
      ue(), R();
    }), (r, o) => (L(), U(De, {
      name: G.value,
      onAfterLeave: ge,
      onAfterEnter: fe,
      persisted: ""
    }, {
      default: oe(() => [
        be(I(
          "button",
          {
            ref_key: "Toaster",
            ref: y,
            class: re(["m-toast", [`--${s.type}`, `--${k.value}`, `--${O.value}`, { "maz-pb-1": s.timeout }]]),
            role: "alert",
            onMouseover: o[2] || (o[2] = (m) => K(!0)),
            onMouseleave: o[3] || (o[3] = (m) => K(!1)),
            onClick: o[4] || (o[4] = M((m) => {
              var z;
              return s.link && !((z = s.link) != null && z.closeToast) ? void 0 : P(m);
            }, ["stop"]))
          },
          [
            b.value ? (L(), U(nt(b.value), {
              key: 0,
              class: "maz-text-2xl"
            })) : D("v-if", !0),
            I("div", Mt, [
              I(
                "p",
                Bt,
                pe(s.message),
                1
                /* TEXT */
              )
            ]),
            s.action ? (L(), U(T(t), {
              key: 1,
              "data-test": "action-btn",
              color: s.type,
              pastel: "",
              loading: B.value,
              size: "sm",
              onClick: o[0] || (o[0] = M((m) => he(s.action.func, m), ["stop"]))
            }, {
              default: oe(() => [
                ot(
                  pe(s.action.text),
                  1
                  /* TEXT */
                )
              ]),
              _: 1
              /* STABLE */
            }, 8, ["color", "loading"])) : D("v-if", !0),
            s.link ? (L(), U(T(t), {
              key: 2,
              "data-test": "link-btn",
              color: s.type,
              pastel: "",
              size: "xs",
              href: s.link.href,
              target: s.link.target ?? "_self"
            }, {
              default: oe(() => {
                var m;
                return [
                  I("div", Ft, [
                    s.link.text ? (L(), Z(
                      "span",
                      Nt,
                      pe(s.link.text),
                      1
                      /* TEXT */
                    )) : D("v-if", !0),
                    ((m = s.link) == null ? void 0 : m.target) == "_blank" ? (L(), U(T(n), {
                      key: 1,
                      class: "maz-text-xl"
                    })) : (L(), U(T(p), {
                      key: 2,
                      class: "maz-text-xl"
                    }))
                  ])
                ];
              }),
              _: 1
              /* STABLE */
            }, 8, ["color", "href", "target"])) : D("v-if", !0),
            s.persistent ? D("v-if", !0) : (L(), Z("button", {
              key: 3,
              class: "--close",
              onClick: o[1] || (o[1] = M((m) => P(m), ["stop"]))
            }, [
              $(T(i), { class: "--icon maz-text-xl" })
            ])),
            I("div", _t, [
              I(
                "div",
                {
                  style: rt({
                    width: J.value
                  }),
                  class: re(["maz-h-full maz-transition-all maz-duration-200 maz-ease-linear", de()])
                },
                null,
                6
                /* CLASS, STYLE */
              )
            ])
          ],
          34
          /* CLASS, NEED_HYDRATION */
        ), [
          [we, F.value]
        ])
      ]),
      _: 1
      /* STABLE */
    }, 8, ["name"]));
  }
}), Ut = /* @__PURE__ */ Re(qt, [["__scopeId", "data-v-ee210f61"]]), Vt = {
  position: "bottom-right",
  timeout: 1e4,
  persistent: !1
};
class jt {
  constructor(e, t) {
    this.app = e, this.globalOptions = t;
  }
  show(e, t) {
    const i = { message: e, ...t }, n = {
      ...Vt,
      ...i,
      ...this.globalOptions,
      ...t
    };
    Me(Ut, {
      props: n,
      app: this.app
    });
  }
  getLocalOptions(e, t) {
    return {
      type: e,
      ...t
    };
  }
  message(e, t) {
    return this.show(e, this.getLocalOptions("theme", t));
  }
  success(e, t) {
    return this.show(e, this.getLocalOptions("success", t));
  }
  error(e, t) {
    return this.show(e, this.getLocalOptions("danger", t));
  }
  info(e, t) {
    return this.show(e, this.getLocalOptions("info", t));
  }
  warning(e, t) {
    return this.show(e, this.getLocalOptions("warning", t));
  }
}
let xe;
const ps = {
  install(s, e) {
    xe = new jt(s, e), s.provide("toast", xe);
  }
}, V = "", Wt = (s) => s.filter((e, t, i) => t === i.indexOf(e)), Rt = (s) => (e = V) => typeof e == "function" ? s.findIndex((...t) => e(...t)) > -1 : s.includes(e), Kt = (s) => s.length > 0, Yt = (s) => (e = V) => Wt([...s, e]), Xt = (s) => (e = V) => s.filter((t) => t !== e);
class Zt {
  constructor() {
    l(this, "_loaders", h([]));
  }
  get loaders() {
    return g(() => this._loaders.value);
  }
  stop(e = V) {
    this._loaders.value = Xt(this._loaders.value)(e);
  }
  start(e = V) {
    return this._loaders.value = Yt(this._loaders.value)(e), () => this.stop(e);
  }
  get anyLoading() {
    return g(() => Kt(this._loaders.value));
  }
  isLoading(e = V) {
    return g(() => Rt(this._loaders.value)(e)).value;
  }
}
const Gt = new Zt(), ys = {
  install: (s) => {
    s.provide("wait", Gt);
  }
}, ye = {
  delay: 100,
  observer: {
    root: void 0,
    rootMargin: "0px",
    threshold: 0.2
  },
  animation: {
    once: !0,
    duration: 300,
    delay: 0
  }
};
class Qt {
  constructor(e) {
    l(this, "options");
    this.options = {
      delay: (e == null ? void 0 : e.delay) ?? ye.delay,
      observer: {
        ...ye.observer,
        ...e == null ? void 0 : e.observer
      },
      animation: {
        ...ye.animation,
        ...e == null ? void 0 : e.animation
      }
    };
  }
  handleIntersect(e, t) {
    for (const i of e) {
      const c = i.target.getAttribute("data-maz-aos-children") === "true", d = i.target.getAttribute("data-maz-aos") ? [i.target] : [];
      if (c) {
        const a = [...document.querySelectorAll("[data-maz-aos-anchor]")].map(
          (v) => v.getAttribute("data-maz-aos-anchor") === `#${i.target.id}` ? v : void 0
        );
        for (const v of a)
          v && d.push(v);
      }
      for (const a of d) {
        const v = a.getAttribute("data-maz-aos-once"), p = typeof v == "string" ? v === "true" : this.options.animation.once;
        if (typeof this.options.observer.threshold == "number" && i.intersectionRatio > this.options.observer.threshold) {
          const y = a.getAttribute("data-maz-aos-duration"), f = a.getAttribute("data-maz-aos-delay");
          if (y || (a.style.transitionDuration = `${this.options.animation.duration}ms`, setTimeout(() => {
            a.style.transitionDuration = "0";
          }, 1e3)), f || (a.style.transitionDelay = `${this.options.animation.delay}ms`, setTimeout(() => {
            a.style.transitionDelay = "0";
          }, 1e3)), a.classList.add("maz-aos-animate"), p) {
            const b = a.getAttribute("data-maz-aos-anchor");
            if (b) {
              const w = document.querySelector(b);
              w && t.unobserve(w);
            }
            t.unobserve(a);
          }
        } else
          a.classList.remove("maz-aos-animate");
      }
    }
  }
  async handleObserver() {
    await X(this.options.delay);
    const e = new IntersectionObserver(this.handleIntersect.bind(this), {
      ...this.options.observer
    });
    for (const t of document.querySelectorAll("[data-maz-aos]")) {
      const i = t.getAttribute("data-maz-aos-anchor");
      if (i) {
        const n = document.querySelector(i);
        n ? (n.setAttribute("data-maz-aos-children", "true"), e.observe(n)) : console.warn(`[maz-ui](aos) no element found with selector "${i}"`);
      } else
        e.observe(t);
    }
  }
  runAnimations() {
    if (S())
      return this.handleObserver();
    console.warn("[MazAos](runAnimations) should be executed on client side");
  }
}
let ne;
const bs = {
  install: (s, e) => {
    ne = new Qt(e), s.provide("aos", ne), S() && (e != null && e.router ? e.router.afterEach(async () => {
      ne.runAnimations();
    }) : ne.runAnimations());
  }
}, Ce = {
  darkClass: "dark",
  lightClass: "light",
  storageThemeKey: "theme",
  storageThemeValueDark: "dark",
  storageThemeValueLight: "light",
  storageThemeValueSystem: "system"
}, W = h("system"), C = h("system");
function Ie({
  darkClass: s,
  lightClass: e,
  storageThemeKey: t,
  storageThemeValueDark: i,
  setLocalStorageValue: n = !0,
  setSelectedTheme: c = !0
}) {
  document.documentElement.classList.remove(e), document.documentElement.classList.add(s), W.value = i, c && (C.value = i), n && (localStorage[t] = i);
}
function Te({
  darkClass: s,
  lightClass: e,
  storageThemeKey: t,
  storageThemeValueLight: i,
  setLocalStorageValue: n = !0,
  setSelectedTheme: c = !0
}) {
  document.documentElement.classList.remove(s), document.documentElement.classList.add(e), W.value = i, c && (C.value = i), n && (localStorage[t] = i);
}
function Ye(s) {
  document.documentElement.classList.remove(s.darkClass), document.documentElement.classList.remove(s.lightClass), W.value = s.storageThemeValueSystem, C.value = s.storageThemeValueSystem, s.setLocalStorageValue && (localStorage[s.storageThemeKey] = s.storageThemeValueSystem), Xe({ ...s, setSelectedTheme: !1 });
}
function Se() {
  return window.matchMedia("(prefers-color-scheme: dark)").matches;
}
function Xe(s) {
  localStorage[s.storageThemeKey] === s.storageThemeValueDark || !(s.storageThemeKey in localStorage) && Se() || localStorage[s.storageThemeKey] === s.storageThemeValueSystem && Se() ? Ie({
    ...s,
    setLocalStorageValue: !1,
    setSelectedTheme: !1
  }) : Te({
    ...s,
    setLocalStorageValue: !1,
    setSelectedTheme: !1
  });
}
function Oe({
  shouldSetDarkMode: s,
  ...e
}) {
  return typeof s != "boolean" ? Ye(e) : s ? Ie(e) : Te(e);
}
function Jt(s) {
  return W.value === s.storageThemeValueDark ? Te(s) : Ie(s);
}
function ws(s = Ce) {
  const e = {
    ...Ce,
    ...s
  }, t = g(() => C.value === e.storageThemeValueDark), i = g(() => C.value === e.storageThemeValueLight), n = g(() => C.value === e.storageThemeValueSystem);
  return Ee(() => {
    localStorage[e.storageThemeKey] && (W.value = localStorage[e.storageThemeKey], C.value = localStorage[e.storageThemeKey]);
  }), {
    autoSetTheme: () => Xe(e),
    toggleTheme: () => Jt(e),
    setSystemTheme: () => Ye({ ...e, setLocalStorageValue: !0 }),
    setDarkTheme: () => Oe({ ...e, shouldSetDarkMode: !0 }),
    setLightTheme: () => Oe({ ...e, shouldSetDarkMode: !1 }),
    hasDarkTheme: t,
    hasLightTheme: i,
    hasSystemTheme: n,
    theme: W,
    selectedTheme: C
  };
}
const zs = ({
  componentName: s,
  instance: e,
  providedId: t
}) => g(() => t ?? `${s}-${e == null ? void 0 : e.uid}`), Es = () => Le("aos"), Ls = () => Le("wait"), Is = () => Le("toast"), Ts = ({
  callback: s,
  options: e
}) => new lt(s, e), ks = ({
  callback: s,
  options: e
}) => new ct(s, e), es = h(), j = h([]), ts = (s, e) => new Promise((t, i) => {
  j.value = [
    ...j.value,
    {
      id: s,
      isActive: !0,
      resolve: async () => {
        await (e == null ? void 0 : e()), t(!0);
      },
      reject: i
    }
  ];
}), Ze = (s) => (j.value = j.value.filter(({ id: e }) => e !== s), j.value), Pe = (s, e, t = !1) => {
  var i;
  e && ((i = e[s]) == null || i.call(e, t), e.isActive = !1, setTimeout(() => {
    Ze(e.id);
  }, 500));
}, As = () => ({
  confirmDialogData: es,
  dialogState: j,
  showDialogAndWaitChoice: ts,
  removeDialogFromState: Ze,
  rejectDialog: (s, e = !1) => Pe("reject", s, e),
  resolveDialog: (s, e = !1) => Pe("resolve", s, e)
});
function ss(s = {}) {
  const {
    internalWindow: e = S() ? window : void 0,
    initialWidth: t = Number.POSITIVE_INFINITY,
    initialHeight: i = Number.POSITIVE_INFINITY,
    includeScrollbar: n = !0
  } = s, c = h(t), d = h(i), a = () => {
    e && (n ? (c.value = e.innerWidth, d.value = e.innerHeight) : (c.value = e.document.documentElement.clientWidth, d.value = e.document.documentElement.clientHeight));
  };
  return a(), e && window.addEventListener("resize", a, { passive: !0 }), { width: c, height: d };
}
function is(s) {
  const e = {};
  for (const [t, i] of Object.entries(s))
    e[t] = Number.parseInt(i, 10);
  return e;
}
function xs({
  initialWidth: s = 0,
  initialHeight: e,
  includeScrollbar: t,
  internalWindow: i,
  listenOrientation: n,
  breakpoints: c,
  mediumBreakPoint: d = "md",
  largeBreakPoint: a = "lg"
}) {
  const { width: v } = ss({
    initialWidth: s,
    // (SSR) mobile first
    initialHeight: e,
    includeScrollbar: t,
    internalWindow: i,
    listenOrientation: n
  }), p = is(c), y = g(() => v.value >= p[a]), f = g(
    () => v.value >= p[d] && v.value < p[a]
  ), b = g(() => v.value >= 0 && v.value < p[d]);
  return {
    width: v,
    numericBreakpoints: p,
    isSmallScreen: b,
    isLargeScreen: y,
    isMediumScreen: f,
    breakpoints: c
  };
}
export {
  Gt as A,
  Qt as B,
  bs as C,
  ne as D,
  ws as E,
  zs as F,
  Es as G,
  Ls as H,
  lt as I,
  Is as J,
  Ts as K,
  ks as L,
  As as M,
  Dt as N,
  xs as O,
  ss as P,
  ls as S,
  jt as T,
  ct as U,
  Zt as W,
  Re as _,
  S as a,
  fs as b,
  cs as c,
  as as d,
  _e as e,
  hs as f,
  je as g,
  ds as h,
  Le as i,
  Ke as j,
  vs as k,
  Ve as l,
  Me as m,
  ms as n,
  gs as o,
  us as p,
  Tt as q,
  Ct as r,
  X as s,
  at as t,
  Ot as u,
  gt as v,
  Ht as w,
  ps as x,
  xe as y,
  ys as z
};
